--- a/src/main/java/org/bukkit/inventory/meta/BookMeta.java
+++ b/src/main/java/org/bukkit/inventory/meta/BookMeta.java
@@ -14,9 +_,6 @@
  * {@link org.bukkit.inventory.ItemStack#getType()}. {@code instanceof} on
  * the meta instance is not sufficient due to unusual inheritance
  * with relation to {@link WritableBookMeta}.
- * <p>
- * <b>Warning:</b> in an upcoming version of Paper, this interface will no
- * longer extend Adventure's {@link net.kyori.adventure.inventory.Book}.
  */
 public interface BookMeta extends WritableBookMeta, net.kyori.adventure.inventory.Book { // Paper - adventure
 
@@ -246,58 +_,27 @@
      */
     void addPages(net.kyori.adventure.text.@NotNull Component @NotNull ... pages);
 
-    /**
-     * @deprecated BookMeta is mutable, there is no need for a builder.
-     */
-    @Deprecated(forRemoval = true, since = "1.21.11")
     interface BookMetaBuilder extends net.kyori.adventure.inventory.Book.Builder {
-        /**
-         * @deprecated BookMeta is mutable, there is no need for a builder.
-         */
-        @Deprecated(forRemoval = true, since = "1.21.11")
+
         @Override
         @NotNull BookMetaBuilder title(net.kyori.adventure.text.@Nullable Component title);
 
-        /**
-         * @deprecated BookMeta is mutable, there is no need for a builder.
-         */
-        @Deprecated(forRemoval = true, since = "1.21.11")
         @Override
         @NotNull BookMetaBuilder author(net.kyori.adventure.text.@Nullable Component author);
 
-        /**
-         * @deprecated BookMeta is mutable, there is no need for a builder.
-         */
-        @Deprecated(forRemoval = true, since = "1.21.11")
         @Override
         @NotNull BookMetaBuilder addPage(net.kyori.adventure.text.@NotNull Component page);
 
-        /**
-         * @deprecated BookMeta is mutable, there is no need for a builder.
-         */
-        @Deprecated(forRemoval = true, since = "1.21.11")
         @Override
         @NotNull BookMetaBuilder pages(net.kyori.adventure.text.@NotNull Component @NotNull ... pages);
 
-        /**
-         * @deprecated BookMeta is mutable, there is no need for a builder.
-         */
-        @Deprecated(forRemoval = true, since = "1.21.11")
         @Override
         @NotNull BookMetaBuilder pages(java.util.@NotNull Collection<net.kyori.adventure.text.Component> pages);
 
-        /**
-         * @deprecated BookMeta is mutable, there is no need for a builder.
-         */
-        @Deprecated(forRemoval = true, since = "1.21.11")
         @Override
         @NotNull BookMeta build();
     }
 
-    /**
-     * @deprecated BookMeta is mutable, there is no need for a builder.
-     */
-    @Deprecated(forRemoval = true, since = "1.21.11")
     @Override
     @NotNull BookMetaBuilder toBuilder();
     // Paper end
