From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 11:08:51 -0600
Subject: [PATCH] Expose getBlockPosBelowThatAffectsMyMovement


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index e24744cc523f1ce3c7c8e27b14d3337a3f17f4bb..95d871a9df97abc338f7f75b3f025385e34248b5 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -1332,4 +1332,12 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
      */
     void broadcastHurtAnimation(@NotNull java.util.Collection<Player> players);
     // Paper end - broadcast hurt animation
+
+    // Slice start
+    /**
+     * Returns the Block that is currently supporting the player, particularly in the case of
+     * crouching over the edge of a block.
+     */
+    org.bukkit.block.Block getBlockStandingOn();
+    // Slice end
 }
