From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 11:58:50 -0600
Subject: [PATCH] Add ObjectId


diff --git a/src/main/java/org/bukkit/entity/HumanEntity.java b/src/main/java/org/bukkit/entity/HumanEntity.java
index ef089ad40e1b994bc4bee52717899937acbd1e27..3303888081eb1561f8e646e5edae47cee5ce6fe8 100644
--- a/src/main/java/org/bukkit/entity/HumanEntity.java
+++ b/src/main/java/org/bukkit/entity/HumanEntity.java
@@ -32,6 +32,11 @@ import org.jspecify.annotations.Nullable;
 @NullMarked
 public interface HumanEntity extends LivingEntity, AnimalTamer, InventoryHolder {
 
+    // Slice start
+    org.bson.types.@Nullable ObjectId getObjectId();
+    void setObjectId(org.bson.types.@Nullable ObjectId objectId);
+    // Slice end
+
     // Paper start
     @Override
     EntityEquipment getEquipment();
