From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 11:08:51 -0600
Subject: [PATCH] Expose getBlockPosBelowThatAffectsMyMovement


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index 2c68aae864a74adf70683eb3b17109aa96572413..073cff2ff325d3a05963acc7c3212e2e4ac38fcc 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -1322,4 +1322,12 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
      */
     void broadcastHurtAnimation(@NotNull java.util.Collection<Player> players);
     // Paper end - broadcast hurt animation
+
+    // Slice start
+    /**
+     * Returns the Block that is currently supporting the player, particularly in the case of
+     * crouching over the edge of a block.
+     */
+    org.bukkit.block.Block getBlockStandingOn();
+    // Slice end
 }
