From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 12 Jan 2025 09:04:30 -0600
Subject: [PATCH] Smooth Teleports


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 0081dd53b6a81ce7892e58d61f9f8a6718e30775..fdb0722b520acb1812fda85b3bfdac0b70fc19dc 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3699,6 +3699,19 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     String getClientBrandName();
     // Paper end
 
+        // Slice start
+    /**
+     * This abuses some of how Minecraft works and allows teleporting a player to another world without
+     * triggering typical respawn packets. All of natural state of chunk resends, entity adds/removes, etc still
+     * happen but the visual "refresh" of a world change is hidden. Depending on the destination location/world,
+     * this can act as a "smooth teleport" to a world if the new world is very similar looking to the old one.
+     *
+     * @param location New location to teleport this Player to
+     */
+    @org.jetbrains.annotations.ApiStatus.Experimental
+    void teleportWithoutRespawn(Location location);
+    // Slice end
+
     // Paper start - Teleport API
     /**
      * Sets the player's rotation.
