From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 12:06:48 -0600
Subject: [PATCH] Add Vanish, settable inv locations


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index 0a5ab13e55c14daedb969adcdc4e0fd664a04bfd..8cd3965a6db91080c54c22d7c7ae28d42bdc2f97 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -1355,4 +1355,20 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
      */
     void setSaveable(boolean saveable);
     // Slice end
+
+
+    // Slice start
+    /**
+     * Returns true if the entity is vanished and cannot emit sounds, effects, etc.
+     */
+    boolean isVanished();
+
+    /**
+     * Sets whether the entity is vanished and cannot emit sounds, effects, etc.
+     *
+     * @param vanished the saveable status
+     * @see #isVanished()
+     */
+    void setVanished(boolean vanished);
+    // Slice end
 }
