From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:36:11 -0600
Subject: [PATCH] Equipment Packet Caching


diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
index ff0d1dd7a1ea3930ba11bd1ce5ea1e1851507bab..0307157471a90176d86762fdfc6350b69ac98ef9 100644
--- a/src/main/java/org/bukkit/entity/LivingEntity.java
+++ b/src/main/java/org/bukkit/entity/LivingEntity.java
@@ -1504,6 +1504,14 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
     void setBodyYaw(float bodyYaw);
     // Paper end - body yaw API
 
+
+    // Slice start
+    /**
+     * @param p The player to send this entity's equipment packet to
+     */
+    void sendEquipment(@NotNull Player p);
+    // Slice end
+
     // Paper start - Expose canUseSlot
     /**
      * Checks whether this entity can use the equipment slot.
