From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:40:26 -0600
Subject: [PATCH] Custom Tracking Ranges


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index 8cd3965a6db91080c54c22d7c7ae28d42bdc2f97..549a89634bfec241f6480ec78ce016853c167309 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -1371,4 +1371,17 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
      */
     void setVanished(boolean vanished);
     // Slice end
+
+    // Slice start
+    /**
+     * Returns the custom tracking range of this Entity
+     */
+    int getTrackingRange();
+
+    /**
+     * Sets the custom tracking range of this Entity. If this is set above 0, it will override any tracking ranges
+     * specified via config.
+     */
+    void setTrackingRange(int range);
+    // Slice end
 }
