From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:54:31 -0600
Subject: [PATCH] World.isShardedFightWorld


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index e2e7cd6c3ab5400830bcc39eff4c11ad821ed0b3..e3b09f76caa257c653af6f9e5dbf3e8dc219d6be 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -2920,6 +2920,11 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
     public void setInstance(boolean value);
     // Slice end
 
+    // Slice start
+    public boolean isShardedFight();
+    public void setShardedFight(boolean value);
+    // Slice end
+
     /**
      * Sets the Difficulty of the world.
      *
