From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:54:31 -0600
Subject: [PATCH] World.isShardedFightWorld


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 21887db41f4262c4ee7e32730e07c15423626d46..8100abcdc441fc0f039ca61ab0c5c5feaec4411c 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -2895,6 +2895,11 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
     public void setInstance(boolean value);
     // Slice end
 
+    // Slice start
+    public boolean isShardedFight();
+    public void setShardedFight(boolean value);
+    // Slice end
+
     /**
      * Sets the Difficulty of the world.
      *
