From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:01:04 -0600
Subject: [PATCH] Add dropLoot option to Block.breakNaturally


diff --git a/src/main/java/org/bukkit/block/Block.java b/src/main/java/org/bukkit/block/Block.java
index 5c75b3d049f22573c4bc85ed06fc7f1efa31b361..d31957eada3d9b938725d6a8a42fe9d9b5620fba 100644
--- a/src/main/java/org/bukkit/block/Block.java
+++ b/src/main/java/org/bukkit/block/Block.java
@@ -570,6 +570,18 @@ public interface Block extends Metadatable, Translatable, net.kyori.adventure.tr
      */
     boolean breakNaturally(boolean triggerEffect, boolean dropExperience);
 
+    // Slice
+    /**
+     * Breaks the block and spawns item drops as if a player had broken it
+     *
+     * @param triggerEffect Play the block break particle effect and sound
+     * @param dropLoot drop loot if the block normally does so
+     * @param dropExperience drop exp if the block normally does so
+     * @return true if the block was destroyed
+     */
+    boolean breakNaturally(boolean triggerEffect, boolean dropLoot, boolean dropExperience);
+    // End Slice
+
     /**
      * Breaks the block and spawns item drops as if a player had broken it
      * with a specific tool
