From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:03:49 -0600
Subject: [PATCH] Add Player.getLastPing


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index c39360b9fdeaa6bf06837300377756b939dcb03d..e6977e43c35b68d1e295a0202a9d7af1e98119a7 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3381,6 +3381,16 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      */
     public int getPing();
 
+    /**
+     * Gets the player's most recent measured ping.
+     *
+     * This differs from {@link #getPing()} as it represents an average of ping over time,
+     * whereas this represents simply the most recent ping.
+     *
+     * @return player's most recent ping
+     */
+    public int getLastPing();
+
     /**
      * Gets the player's current locale.
      *
