From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:03:49 -0600
Subject: [PATCH] Add Player.getLastPing


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 91e59c154d0013952059700edf7e2840aef1b724..ecd74fc252b62f87d9ab92c5fb70761ef3ca34ad 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3376,6 +3376,16 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      */
     public int getPing();
 
+    /**
+     * Gets the player's most recent measured ping.
+     *
+     * This differs from {@link #getPing()} as it represents an average of ping over time,
+     * whereas this represents simply the most recent ping.
+     *
+     * @return player's most recent ping
+     */
+    public int getLastPing();
+
     /**
      * Gets the player's current locale.
      *
