From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:39:44 -0600
Subject: [PATCH] Player AFK


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index e6977e43c35b68d1e295a0202a9d7af1e98119a7..34932dd9e9a4da3480c1fb3d851eda9adcefc34d 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -4043,4 +4043,16 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     default ObjectContents asObjectContents() {
         return this.getPlayerProfile().asObjectContents();
     }
+
+    /**
+     * @return Whether this player is AFK.
+     */
+    // Slice start
+    boolean isAfk();
+
+    /**
+     * @param afk whether the player is AFK.
+     */
+    void setAfk(boolean afk);
+    // Slice end
 }
