From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:39:44 -0600
Subject: [PATCH] Player AFK


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index ecd74fc252b62f87d9ab92c5fb70761ef3ca34ad..7473f0d45626b122e0681d05c55488cfdd915d93 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -4035,4 +4035,16 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      */
     @ApiStatus.Experimental
     PlayerGameConnection getConnection();
+
+    /**
+     * @return Whether this player is AFK.
+     */
+    // Slice start
+    boolean isAfk();
+
+    /**
+     * @param afk whether the player is AFK.
+     */
+    void setAfk(boolean afk);
+    // Slice end
 }
