From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 21 Jan 2025 15:00:18 -0600
Subject: [PATCH] Add Vanish


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index f2f33847133c77341bae1d51f7ff2cb222d9bf4b..4f4e6b00c22f3c702741446bde30aa7115136d79 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -1284,4 +1284,20 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
      */
     void setSaveable(boolean saveable);
     // Slice end
+
+
+    // Slice start
+    /**
+     * Returns true if the entity is vanished and cannot emit sounds, effects, etc.
+     */
+    boolean isVanished();
+
+    /**
+     * Sets whether the entity is vanished and cannot emit sounds, effects, etc.
+     *
+     * @param vanished the saveable status
+     * @see #isVanished()
+     */
+    void setVanished(boolean vanished);
+    // Slice end
 }
