From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 21 Jan 2026 08:54:18 -0600
Subject: [PATCH] Add direct remover to HangingBreakByEntityEvent


diff --git a/src/main/java/org/bukkit/event/hanging/HangingBreakByEntityEvent.java b/src/main/java/org/bukkit/event/hanging/HangingBreakByEntityEvent.java
index 1a5d9d99330cc63dc271ec6499019990ac950ff1..8d5661fe9a608e19908fc499fd67c11d25a68848 100644
--- a/src/main/java/org/bukkit/event/hanging/HangingBreakByEntityEvent.java
+++ b/src/main/java/org/bukkit/event/hanging/HangingBreakByEntityEvent.java
@@ -5,6 +5,7 @@ import org.bukkit.entity.Entity;
 import org.bukkit.entity.Hanging;
 import org.jetbrains.annotations.ApiStatus;
 import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
 
 /**
  * Triggered when a hanging entity is removed by an entity
@@ -14,18 +15,27 @@ public class HangingBreakByEntityEvent extends HangingBreakEvent {
 
     private final Entity remover;
     private final DamageSource damageSource;
+    private final @Nullable Entity directRemover; // Slice
 
     @ApiStatus.Internal
     public HangingBreakByEntityEvent(final Hanging hanging, final Entity remover, final DamageSource damageSource) {
-        this(hanging, remover, damageSource, HangingBreakEvent.RemoveCause.ENTITY);
+        this(hanging, remover, damageSource, null, HangingBreakEvent.RemoveCause.ENTITY); // Slice
     }
 
     @ApiStatus.Internal
     public HangingBreakByEntityEvent(final Hanging hanging, final Entity remover, final DamageSource damageSource, final HangingBreakEvent.RemoveCause cause) {
+        this(hanging, remover, damageSource, null, cause); // Slice
+    }
+
+    // Slice start
+    @ApiStatus.Internal
+    public HangingBreakByEntityEvent(final Hanging hanging, final Entity remover, final DamageSource damageSource, final @Nullable Entity directRemover, final HangingBreakEvent.RemoveCause cause) {
         super(hanging, cause);
         this.remover = remover;
         this.damageSource = damageSource;
+        this.directRemover = directRemover;
     }
+    // Slice end
 
     /**
      * Gets the entity that removed the hanging entity.
@@ -44,4 +54,15 @@ public class HangingBreakByEntityEvent extends HangingBreakEvent {
     public DamageSource getDamageSource() {
         return this.damageSource;
     }
+
+    // Slice start
+    /**
+     * Gets the direct entity that removed the hanging entity. For example, a thrown ender pearl.
+     *
+     * @return the entity that removed the hanging entity
+     */
+    public @Nullable Entity getDirectRemover() {
+        return this.directRemover;
+    }
+    // Slice end
 }
