From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 22 Jul 2025 13:57:39 -0500
Subject: [PATCH] World.isShardedFightWorld


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 178386708819d194f4eb2964fc6d5e4865297dcf..8ffcb797fbc55eaa5869c85f96a9251181d86e96 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -27,7 +27,6 @@ import org.bukkit.entity.SpawnCategory;
 import org.bukkit.event.entity.CreatureSpawnEvent;
 import org.bukkit.generator.BiomeProvider;
 import org.bukkit.generator.BlockPopulator;
-import org.bukkit.generator.ChunkGenerator;
 import org.bukkit.generator.WorldInfo;
 import org.bukkit.generator.structure.GeneratedStructure;
 import org.bukkit.generator.structure.Structure;
@@ -2838,6 +2837,11 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
     public void setInstance(boolean value);
     // Slice end
 
+    // Slice start
+    public boolean isShardedFight();
+    public void setShardedFight(boolean value);
+    // Slice end
+
     /**
      * Sets the Difficulty of the world.
      *
