From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 22 Jul 2025 13:57:39 -0500
Subject: [PATCH] World.isShardedFightWorld


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 3cc75759390250a6d379d8eede778c40d788e43e..89d85f87cd62f264be0c95deaa87a4ee607148b1 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -28,7 +28,6 @@ import org.bukkit.entity.SpawnCategory;
 import org.bukkit.event.entity.CreatureSpawnEvent;
 import org.bukkit.generator.BiomeProvider;
 import org.bukkit.generator.BlockPopulator;
-import org.bukkit.generator.ChunkGenerator;
 import org.bukkit.generator.WorldInfo;
 import org.bukkit.generator.structure.GeneratedStructure;
 import org.bukkit.generator.structure.Structure;
@@ -2814,6 +2813,11 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
     public void setInstance(boolean value);
     // Slice end
 
+    // Slice start
+    public boolean isShardedFight();
+    public void setShardedFight(boolean value);
+    // Slice end
+
     /**
      * Sets the Difficulty of the world.
      *
