--- a/src/main/java/io/papermc/paper/command/brigadier/argument/WrappedArgumentCommandNode.java
+++ b/src/main/java/io/papermc/paper/command/brigadier/argument/WrappedArgumentCommandNode.java
@@ -46,7 +_,7 @@
     @Override
     public void parse(final StringReader reader, final CommandContextBuilder<CommandSourceStack> contextBuilder) throws CommandSyntaxException {
         final int start = reader.getCursor();
-        final API result = this.pureArgumentType.parse(reader, contextBuilder.getSource()); // Use the api argument parser
+        final API result = this.pureArgumentType.parse(reader); // Use the api argument parser
         final ParsedArgument<CommandSourceStack, API> parsed = new ParsedArgument<>(start, reader.getCursor(), result); // Return an API parsed argument instead.
 
         contextBuilder.withArgument(this.getName(), parsed);
