--- a/src/main/java/com/destroystokyo/paper/event/server/AsyncTabCompleteEvent.java
+++ b/src/main/java/com/destroystokyo/paper/event/server/AsyncTabCompleteEvent.java
@@ -240,8 +_,6 @@
 
     /**
      * A rich tab completion, consisting of a string suggestion, and a nullable {@link Component} tooltip.
-     * <p>
-     * <b>Warning:</b> In a future update, this class will no longer implement {@link Examinable}.
      */
     public interface Completion extends Examinable {
 
@@ -260,7 +_,6 @@
         @Nullable Component tooltip();
 
         @Override
-        @Deprecated(forRemoval = true, since = "1.21.11")
         default Stream<? extends ExaminableProperty> examinableProperties() {
             return Stream.of(ExaminableProperty.of("suggestion", this.suggestion()), ExaminableProperty.of("tooltip", this.tooltip()));
         }
