--- a/src/main/java/org/bukkit/Registry.java
+++ b/src/main/java/org/bukkit/Registry.java
@@ -3,7 +_,6 @@
 import com.google.common.base.Preconditions;
 import com.google.common.base.Predicates;
 import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.Iterables;
 import io.papermc.paper.datacomponent.DataComponentType;
 import io.papermc.paper.registry.RegistryAccess;
 import io.papermc.paper.registry.RegistryKey;
@@ -299,11 +_,6 @@
         }
 
         @Override
-        public int size() {
-            return MemoryKey.values().size();
-        }
-
-        @Override
         public @Nullable MemoryKey get(final NamespacedKey key) {
             return MemoryKey.getByKey(key);
         }
@@ -542,13 +_,6 @@
         return (namespacedKey != null) ? this.get(namespacedKey) : null;
     }
 
-    /**
-     * Gets the size of the registry.
-     *
-     * @return the size of the registry
-     */
-    int size();
-
     @ApiStatus.Internal
     class SimpleRegistry<T extends Enum<T> & Keyed> extends NotARegistry<T> { // Paper - remove final
 
@@ -578,11 +_,6 @@
         }
 
         @Override
-        public int size() {
-            return map.size();
-        }
-
-        @Override
         public Iterator<T> iterator() {
             return this.map.values().iterator();
         }
@@ -600,11 +_,6 @@
         @Override
         public Stream<A> stream() {
             return StreamSupport.stream(this.spliterator(), false);
-        }
-
-        @Override
-        public int size() {
-            return Iterables.size(this);
         }
 
         @Override
