--- a/src/main/java/io/papermc/paper/configuration/RemovedConfigurations.java
+++ b/src/main/java/io/papermc/paper/configuration/RemovedConfigurations.java
@@ -54,8 +_,7 @@
         path("fixes", "fix-curing-zombie-villager-discount-exploit"),
         path("entities", "mob-effects", "undead-immune-to-certain-effects"),
         path("entities", "entities-target-with-follow-range"),
-        path("environment", "disable-teleportation-suffocation-check"),
-        path("misc", "light-queue-size")
+        path("environment", "disable-teleportation-suffocation-check")
     };
     // spawn.keep-spawn-loaded and spawn.keep-spawn-loaded-range are no longer used, but kept
     // in the world default config for compatibility with old worlds being migrated to use the gamerule
