--- a/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
+++ b/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
@@ -327,9 +_,6 @@
                     public int day = 5;
                 }
             }
-
-            @Comment("Adds a cooldown to bees being released after a failed release, which can occur if the hive is blocked or it being night.")
-            public boolean cooldownFailedBeehiveReleases = true;
         }
 
         public TrackingRangeY trackingRangeY;
@@ -558,6 +_,7 @@
     public Misc misc;
 
     public class Misc extends ConfigurationPart {
+        public int lightQueueSize = 20;
         public boolean updatePathfindingOnBlockUpdate = true;
         public boolean showSignClickCommandFailureMsgsToPlayer = false;
         public RedstoneImplementation redstoneImplementation = RedstoneImplementation.VANILLA;
