--- a/net/minecraft/world/item/MobBucketItem.java
+++ b/net/minecraft/world/item/MobBucketItem.java
@@ -58,7 +_,7 @@
         }
 
         if (mob != null) {
-            serverLevel.addFreshEntityWithPassengers(mob, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.BUCKET); // Paper - Add SpawnReason
+            serverLevel.addFreshEntityWithPassengers(mob);
             mob.playAmbientSound();
         }
     }
