From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 11:07:32 -0600
Subject: [PATCH] Allow opening covered chests


diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftEnderChest.java b/src/main/java/org/bukkit/craftbukkit/block/CraftEnderChest.java
index 2a9243e416bb75f8e942b08a7d3d22bd6ba4df47..60292ee5902ac08010cadbb6e0dfde373bb12b6a 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/CraftEnderChest.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftEnderChest.java
@@ -50,9 +50,7 @@ public class CraftEnderChest extends CraftBlockEntityState<EnderChestBlockEntity
 
     @Override
     public boolean isBlocked() {
-        // Uses the same logic as EnderChestBlock's check for opening container
-        final BlockPos abovePos = this.getPosition().above();
-        return this.isPlaced() && this.getWorldHandle().getBlockState(abovePos).isRedstoneConductor(this.getWorldHandle(), abovePos);
+        return false; // Slice - Allow opening covered chests
     }
 
     @Override
