From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 11:08:51 -0600
Subject: [PATCH] Expose getBlockPosBelowThatAffectsMyMovement


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index e45f9514e4b593111e827a2fc083381aa8072f67..59c640af00ecd757440c5eaca721c388a4fd3948 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1339,4 +1339,11 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         return this.entity.get(io.papermc.paper.datacomponent.PaperDataComponentType.bukkitToMinecraft(type)) != null;
     }
 
+    // Slice start
+    @Override
+    public org.bukkit.block.Block getBlockStandingOn() {
+        net.minecraft.core.BlockPos pos = this.entity.getBlockPosBelowThatAffectsMyMovement();
+        return this.entity.level().getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ());
+    }
+    // Slice end
 }
