From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 11:58:50 -0600
Subject: [PATCH] Add ObjectId


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
index 2aea80487ff1bc68e4aff21909658e307c0da956..5f62db05db09d7e47465b77403b4ab5338225cdc 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
@@ -84,6 +84,7 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
     protected final PermissibleBase perm = new PermissibleBase(this);
     private boolean op;
     private GameMode mode;
+    private org.bson.types.ObjectId objectId;
 
     public CraftHumanEntity(final CraftServer server, final Player entity) {
         super(server, entity);
@@ -107,6 +108,18 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
         return this.getClass().getSimpleName() + "{name=" + this.getName() + ", uuid=" + this.getUniqueId() + '}';
     }
 
+    // Slice start
+    @Override
+    public org.bson.types.ObjectId getObjectId() {
+        return objectId;
+    }
+
+    @Override
+    public void setObjectId(org.bson.types.ObjectId objectId) {
+        this.objectId = objectId;
+    }
+    // Slice end
+
     @Override
     public PlayerInventory getInventory() {
         return this.inventory;
