From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 3 Aug 2025 12:07:20 -0500
Subject: [PATCH] Packet obfuscation and reduction


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 33ee20c83665bc359d1d45586d6882953a8660a1..0f9160a9b1fe3901250063856b3146aff34d4114 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2765,7 +2765,8 @@ public class CraftPlayer extends CraftHumanEntity implements Player, PluginMessa
                 this.sendHealthUpdate();
             }
         }
-        this.getHandle().getEntityData().set(net.minecraft.world.entity.LivingEntity.DATA_HEALTH_ID, (float) this.getScaledHealth());
+        float scaledHealth = this.getScaledHealth();
+        this.getHandle().getEntityData().set(net.minecraft.world.entity.LivingEntity.DATA_HEALTH_ID, scaledHealth, scaledHealth <= 0 ? 0f : 20f); // Slice
 
         this.getHandle().maxHealthCache = this.getMaxHealth();
     }
