From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 21 Jan 2025 10:58:41 -0600
Subject: [PATCH] Add ObjectId


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
index a396157548a5b3c3e86206c35789bb40346c701c..c21db3810a30b5ac98253d295725f6adb079b76d 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
@@ -81,6 +81,7 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
     protected final PermissibleBase perm = new PermissibleBase(this);
     private boolean op;
     private GameMode mode;
+    private org.bson.types.ObjectId objectId;
 
     public CraftHumanEntity(final CraftServer server, final Player entity) {
         super(server, entity);
@@ -89,6 +90,18 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
         this.enderChest = new CraftInventory(entity.getEnderChestInventory());
     }
 
+    // Slice start
+    @Override
+    public org.bson.types.ObjectId getObjectId() {
+        return objectId;
+    }
+
+    @Override
+    public void setObjectId(org.bson.types.ObjectId objectId) {
+        this.objectId = objectId;
+    }
+    // Slice end
+
     @Override
     public PlayerInventory getInventory() {
         return this.inventory;
