From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 19 Jan 2025 08:35:56 -0600
Subject: [PATCH] Expose getBlockPosBelowThatAffectsMyMovement


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index 6d81a19741868983c54aff6c2c4c0e2bf690ba0d..8c35e3fe9e7e9749bde334b275bb7e23d3f3c732 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1333,4 +1333,12 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         }
     }
     // Paper end - broadcast hurt animation
+
+    // Slice start
+    @Override
+    public org.bukkit.block.Block getBlockStandingOn() {
+        net.minecraft.core.BlockPos pos = this.entity.getBlockPosBelowThatAffectsMyMovement();
+        return this.entity.level().getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ());
+    }
+    // Slice end
 }
