From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 19 Jan 2025 09:13:05 -0600
Subject: [PATCH] Non-saveable-entities


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index 8c35e3fe9e7e9749bde334b275bb7e23d3f3c732..98b983cbc41ab95dda327070237ff040182334c7 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1341,4 +1341,16 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         return this.entity.level().getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ());
     }
     // Slice end
+
+    // Slice start
+    @Override
+    public boolean isSaveable() {
+        return this.entity.saveable;
+    }
+
+    @Override
+    public void setSaveable(boolean saveable) {
+        this.entity.saveable = saveable;
+    }
+    // Slice end
 }
