From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:50:48 -0600
Subject: [PATCH] Expose Entity.despawnTime


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index c0721c41e86b04e3d10d0d2e3c136ae0402c340f..6ae9ca3f1248e9121072880336fccd5424add636 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1374,4 +1374,16 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         this.entity.trackingRange = range;
     }
     // Slice end
+
+    // Slice start
+    @Override
+    public int getDespawnTime() {
+        return this.entity.despawnTime;
+    }
+
+    @Override
+    public void setDespawnTime(int ticks) {
+        this.entity.despawnTime = ticks;
+    }
+    // Slice end
 }
