From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:54:28 -0600
Subject: [PATCH] World.isShardedFightWorld


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index f59c3a6a5421b1ab31063b96ef71bbd8f2817bf7..796f850dc6a202bc829043dbb095b4db39dd54d4 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -1165,6 +1165,18 @@ public class CraftWorld extends CraftRegionAccessor implements World {
     }
     // Slice end
 
+    // Slice start
+    @Override
+    public boolean isShardedFight() {
+        return world.shardedFight;
+    }
+
+    @Override
+    public void setShardedFight(boolean value) {
+        world.shardedFight = value;
+    }
+    // Slice end
+
     @Override
     public void setDifficulty(Difficulty difficulty) {
         this.getHandle().getServer().setDifficulty(this.getHandle(), CraftDifficulty.toMinecraft(difficulty), null, true);
