From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:01:00 -0600
Subject: [PATCH] Add dropLoot option to Block.breakNaturally


diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
index afbf9c70f69b02e7b92a8382aa9fb9f1132b7bf8..2ad6abca3d52d1cf1653f92b7f681c36b823983a 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
@@ -454,12 +454,24 @@ public class CraftBlock implements Block {
     }
 
     @Override
+    // Slice start
+    public boolean breakNaturally(boolean triggerEffect, boolean dropLoot, boolean dropExperience) {
+        return this.breakNaturally(null, triggerEffect, dropExperience, triggerEffect, dropLoot);
+    }
+
+    @Override
+    // Slice end
     public boolean breakNaturally(ItemStack item, boolean triggerEffect, boolean dropExperience) {
         return this.breakNaturally(item, triggerEffect, dropExperience, false);
     }
 
     @Override
     public boolean breakNaturally(ItemStack item, boolean triggerEffect, boolean dropExperience, boolean forceEffect) {
+        // Slice start
+        return this.breakNaturally(item, triggerEffect, dropExperience, forceEffect, true);
+    }
+
+    public boolean breakNaturally(ItemStack item, boolean triggerEffect, boolean dropExperience, boolean forceEffect, boolean dropLoot) { // Slice end
         // Order matters here, need to drop before setting to air so skulls can get their data
         BlockState state = this.getBlockState();
         net.minecraft.world.level.block.Block block = state.getBlock();
@@ -467,7 +479,7 @@ public class CraftBlock implements Block {
         boolean result = false;
 
         // Modelled off Player#hasCorrectToolForDrops
-        if (!state.isAir() && (item == null || !state.requiresCorrectToolForDrops() || nmsItem.isCorrectToolForDrops(state))) {
+        if (!state.isAir() && dropLoot && (item == null || !state.requiresCorrectToolForDrops() || nmsItem.isCorrectToolForDrops(state))) { // Slice - add dropLoot
             net.minecraft.world.level.block.Block.dropResources(state, this.level.getMinecraftWorld(), this.position, this.level.getBlockEntity(this.position), null, nmsItem, false);
             if (dropExperience) block.popExperience(this.level.getMinecraftWorld(), this.position, block.getExpDrop(state, this.level.getMinecraftWorld(), this.position, nmsItem, true));
             result = true;
