From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:03:45 -0600
Subject: [PATCH] Add Player.getLastPing


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index b2e66948561fbd483af5fb7712dee2d47e0c11e5..1a8a9c90c7c0a4e0cf1c866658ec770844e55551 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2713,6 +2713,14 @@ public class CraftPlayer extends CraftHumanEntity implements Player, PluginMessa
         return this.getHandle().connection.latency();
     }
 
+    // Paper start
+    @Override
+    public int getLastPing() {
+        if (this.getHandle().connection == null) throw new UnsupportedOperationException("Too early to call this method at this stage");
+        return this.getHandle().connection.lastLatency();
+    }
+    // Paper end
+
     @Override
     public String getLocale() {
         // Paper start - Locale change event
