From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:03:45 -0600
Subject: [PATCH] Add Player.getLastPing


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 66d5db455d02e13b0e8c7b3ce25fb922a7609de3..5ce583e93366d856632d4af31c1c7715556ef7a8 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2738,6 +2738,14 @@ public class CraftPlayer extends CraftHumanEntity implements Player, PluginMessa
         return this.getHandle().connection.latency();
     }
 
+    // Paper start
+    @Override
+    public int getLastPing() {
+        if (this.getHandle().connection == null) throw new UnsupportedOperationException("Too early to call this method at this stage");
+        return this.getHandle().connection.lastLatency();
+    }
+    // Paper end
+
     @Override
     public String getLocale() {
         // Paper start - Locale change event
