From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:29:27 -0600
Subject: [PATCH] Packet obfuscation and reduction


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 2d45a6240f189a85e77fb4bc0005c7908cd4715e..ef0ff9c6e67f948af249378961b4e80cbb7ed7fd 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2614,7 +2614,8 @@ public class CraftPlayer extends CraftHumanEntity implements Player, PluginMessa
                 this.sendHealthUpdate();
             }
         }
-        this.getHandle().getEntityData().set(net.minecraft.world.entity.LivingEntity.DATA_HEALTH_ID, (float) this.getScaledHealth());
+        float scaledHealth = this.getScaledHealth();
+        this.getHandle().getEntityData().set(net.minecraft.world.entity.LivingEntity.DATA_HEALTH_ID, scaledHealth, scaledHealth <= 0 ? 0f : 20f); // Slice
 
         this.getHandle().maxHealthCache = this.getMaxHealth();
     }
