From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 21 Jan 2025 15:00:18 -0600
Subject: [PATCH] Add Vanish


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index e02347741c48280e0d70e5dd3f2d992827b4a1ce..136965d6f74e5e686b9bfab545533b907dedf09a 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1353,4 +1353,16 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         this.entity.saveable = saveable;
     }
     // Slice end
+
+    // Slice start
+    @Override
+    public boolean isVanished() {
+        return this.entity.vanished;
+    }
+
+    @Override
+    public void setVanished(boolean vanished) {
+        this.entity.vanished = vanished;
+    }
+    // Slice end
 }
diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventoryCustom.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventoryCustom.java
index 393e44cd3ed8d9f43b0ab42dbf2f9e82e8512dbd..d89f406b4b8f26cf9e5407d43eb3ef3754f77e3f 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventoryCustom.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventoryCustom.java
@@ -69,6 +69,14 @@ public class CraftInventoryCustom extends CraftInventory {
     }
     // Paper end
 
+    // Slice start
+    public void setLocation(Location location) {
+        if (this.inventory instanceof MinecraftInventory minecraftInventory) {
+            minecraftInventory.location = location;
+        }
+    }
+    // Slice end
+
     static class MinecraftInventory implements Container {
         private final NonNullList<ItemStack> items;
         private int maxStack = MAX_STACK;
@@ -77,6 +85,7 @@ public class CraftInventoryCustom extends CraftInventory {
         private final net.kyori.adventure.text.Component adventure$title; // Paper
         private InventoryType type;
         private final InventoryHolder owner;
+        private Location location; // Slice
 
         // Paper start
         public MinecraftInventory(InventoryHolder owner, InventoryType type, net.kyori.adventure.text.Component title) {
@@ -239,7 +248,7 @@ public class CraftInventoryCustom extends CraftInventory {
 
         @Override
         public Location getLocation() {
-            return null;
+            return location;
         }
 
         // Paper start
