From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:39:40 -0600
Subject: [PATCH] Player AFK


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index ef0ff9c6e67f948af249378961b4e80cbb7ed7fd..afcc534098be197b188da4991fff9770736d85ab 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -3325,6 +3325,16 @@ public class CraftPlayer extends CraftHumanEntity implements Player, PluginMessa
         return this.getHandle().connection.playerGameConnection;
     }
 
+    @Override
+    public boolean isAfk() {
+        return getHandle().afk;
+    }
+
+    @Override
+    public void setAfk(boolean afk) {
+        getHandle().afk = afk;
+    }
+
     @Override
     public org.bukkit.entity.Entity releaseLeftShoulderEntity() {
         if (!getHandle().getShoulderEntityLeft().isEmpty()) {
