From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 27 Apr 2025 08:18:32 -0500
Subject: [PATCH] Set multiple team settings at once


diff --git a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java
index 2b335c750ce5f9ccc2651a8701497ca9b8f46704..c508b2003cb7980b41dce542e5c3053b2fb9d1d8 100644
--- a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java
+++ b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java
@@ -341,6 +341,37 @@ final class CraftTeam extends CraftScoreboardComponent implements Team {
         }
     }
 
+    //Slice start
+    @Override
+    public void teamOptions(net.kyori.adventure.text.Component displayName,
+                            net.kyori.adventure.text.Component prefix,
+                            net.kyori.adventure.text.Component suffix,
+                            net.kyori.adventure.text.format.NamedTextColor color,
+                            java.util.List<org.apache.commons.lang3.tuple.Pair<Option, OptionStatus>> options) throws IllegalStateException {
+        checkState();
+        team.setUpdateTeamChangeInstantly(false);
+
+        if (displayName == null) displayName = net.kyori.adventure.text.Component.empty();
+        team.setDisplayName(io.papermc.paper.adventure.PaperAdventure.asVanilla(displayName));
+
+        if (prefix == null) prefix = net.kyori.adventure.text.Component.empty();
+        team.setPlayerPrefix(io.papermc.paper.adventure.PaperAdventure.asVanilla(prefix));
+
+        if (suffix == null) suffix = net.kyori.adventure.text.Component.empty();
+        team.setPlayerSuffix(io.papermc.paper.adventure.PaperAdventure.asVanilla(suffix));
+
+        if (color == null) color = net.kyori.adventure.text.format.NamedTextColor.WHITE;
+        team.setColor(io.papermc.paper.adventure.PaperAdventure.asVanilla(color));
+
+        for (org.apache.commons.lang3.tuple.Pair<Option, OptionStatus> pair : options) {
+            setOption(pair.getLeft(), pair.getRight());
+        }
+
+        team.setUpdateTeamChangeInstantly(true);
+        team.getScoreboard().onTeamChanged(team);
+    }
+    //Slice end
+
     // Paper start
     @Override
     public void addEntity(org.bukkit.entity.Entity entity) throws IllegalStateException, IllegalArgumentException {
