From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 8 Jun 2025 11:18:17 -0500
Subject: [PATCH] Player AFK


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 4ea7488194a57e1620a2428281dbbde54c1705d3..7215ee45a9fafc53c61ff08b2cdd5ffac4a600ad 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -3609,4 +3609,14 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
         return forwardMovement == backwardMovement ? 0 : forwardMovement ? 1 : -1;
     }
+
+    @Override
+    public boolean isAfk() {
+        return getHandle().afk;
+    }
+
+    @Override
+    public void setAfk(boolean afk) {
+        getHandle().afk = afk;
+    }
 }
