From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 11 May 2025 08:03:48 -0500
Subject: [PATCH] Expose Entity.despawnTime


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index dcaa766e1561c922972096ee4b566e1282e49260..c53f2544a5ac8ea0fa3e7b460835bbc0d1586b6b 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1377,4 +1377,16 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         this.entity.trackingRange = range;
     }
     // Slice end
+
+    // Slice start
+    @Override
+    public int getDespawnTime() {
+        return this.entity.despawnTime;
+    }
+
+    @Override
+    public void setDespawnTime(int ticks) {
+        this.entity.despawnTime = ticks;
+    }
+    // Slice end
 }
