From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 22 Jul 2025 13:57:39 -0500
Subject: [PATCH] World.isShardedFightWorld


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index 7c49a0a70e858331cfeb6a69b64a289872e94e2d..83dcf9274e8fe8805fea4d9d5b470513e48dd5d0 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -1297,6 +1297,18 @@ public class CraftWorld extends CraftRegionAccessor implements World {
     }
     // Slice end
 
+    // Slice start
+    @Override
+    public boolean isShardedFight() {
+        return world.shardedFight;
+    }
+
+    @Override
+    public void setShardedFight(boolean value) {
+        world.shardedFight = value;
+    }
+    // Slice end
+
     @Override
     public void setDifficulty(Difficulty difficulty) {
         this.getHandle().getServer().setDifficulty(this.getHandle(), net.minecraft.world.Difficulty.byId(difficulty.getValue()), true); // Paper - per level difficulty; don't skip other difficulty-changing logic
