--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -2214,11 +_,7 @@
             if (invulnerabilityRelatedLastDamage == 0) return 0D; // no last damage, no reduction
             // last damage existed, this means the reduction *technically* is (new damage - last damage).
             // If the event damage was changed to something less than invul damage, hard lock it at 0.
-            //
-            // Cast the passed in double down to a float as double -> float -> double is lossy.
-            // If last damage is a (float) 3.2D (since the events use doubles), we cannot compare
-            // the new damage value of this damage instance by upcasting it again to a double as 3.2D != (double) (float) 3.2D.
-            if (d.floatValue() < invulnerabilityRelatedLastDamage) return 0D;
+            if (d < invulnerabilityRelatedLastDamage) return 0D;
             return (double) -invulnerabilityRelatedLastDamage;
         };
         final float originalInvulnerabilityReduction = invulnerabilityReductionEquation.apply((double) amount).floatValue();
@@ -2347,10 +_,7 @@
 
             // Apply damage to armor
             if (!damageSource.is(DamageTypeTags.BYPASSES_ARMOR)) {
-                float armorDamage = (float) event.getDamage();
-                armorDamage += (float) event.getDamage(DamageModifier.INVULNERABILITY_REDUCTION);
-                armorDamage += (float) event.getDamage(DamageModifier.BLOCKING);
-                armorDamage += (float) event.getDamage(DamageModifier.HARD_HAT);
+                float armorDamage = (float) (event.getDamage() + event.getDamage(DamageModifier.BLOCKING) + event.getDamage(DamageModifier.HARD_HAT));
                 this.hurtArmor(damageSource, armorDamage);
             }
 
