--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -1125,9 +_,7 @@
                     double d = this.getEquipmentDropChance(equipmentSlot);
                     if (d > 1.0) {
                         this.setItemSlot(equipmentSlot, ItemStack.EMPTY);
-                        this.forceDrops = true; // Paper - Add missing forceDrop toggles
                         this.spawnAtLocation(level, itemBySlot);
-                        this.forceDrops = false; // Paper - Add missing forceDrop toggles
                     }
                 }
             }
