--- a/net/minecraft/network/FriendlyByteBuf.java
+++ b/net/minecraft/network/FriendlyByteBuf.java
@@ -77,10 +_,6 @@
     private static final int MAX_PUBLIC_KEY_HEADER_SIZE = 256;
     private static final int MAX_PUBLIC_KEY_LENGTH = 512;
     private static final Gson GSON = new Gson();
-    // Paper start - Track codec depth
-    public boolean trackCodecDepth;
-    public byte codecDepth;
-    // Paper end - Track codec depth
 
     public FriendlyByteBuf(ByteBuf source) {
         this.adventure$locale = PacketEncoder.ADVENTURE_LOCALE.get(); // Paper - track player's locale for server-side translations
