From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 12 Jan 2025 09:02:34 -0600
Subject: [PATCH] Add Player to SpongeAbsorbEvent


diff --git a/net/minecraft/world/level/block/SpongeBlock.java b/net/minecraft/world/level/block/SpongeBlock.java
index 88e7320a48b08f052ae1cac914ebbcb905baf693..5eb5fefa8b623cf82aec69f1fe9c794bb3ff297c 100644
--- a/net/minecraft/world/level/block/SpongeBlock.java
+++ b/net/minecraft/world/level/block/SpongeBlock.java
@@ -102,7 +102,8 @@ public class SpongeBlock extends Block {
         if (!blocks.isEmpty()) {
             final org.bukkit.block.Block sponge = org.bukkit.craftbukkit.block.CraftBlock.at(level, pos);
 
-            org.bukkit.event.block.SpongeAbsorbEvent event = new org.bukkit.event.block.SpongeAbsorbEvent(sponge, (java.util.List<org.bukkit.block.BlockState>) (java.util.List) blocks);
+            Level.PendingBlockEvent blockEvent = level.pendingPlayerBlockEvents.remove(pos); // Paper
+            org.bukkit.event.block.SpongeAbsorbEvent event = new org.bukkit.event.block.SpongeAbsorbEvent(sponge, blockEvent != null ? (org.bukkit.entity.Player) blockEvent.player.getBukkitEntity() : null, (java.util.List<org.bukkit.block.BlockState>) (java.util.List) blocks);
             if (!event.callEvent()) {
                 return false;
             }
