From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 11:06:24 -0600
Subject: [PATCH] Add Force Crit to PlayerPreAttackEntityEvent


diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index 5d4b3ed09fe10f545d0720b2919f0d6c822d5196..742e6d2d3ae7d514cc32d71c2ae026d40f4b8cf2 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -1049,7 +1049,7 @@ public abstract class Player extends Avatar implements ContainerUser {
                     }
 
                     baseDamage += attackingItemStack.getItem().getAttackDamageBonus(entity, baseDamage, damageSource);
-                    boolean criticalAttack = fullStrengthAttack && this.canCriticalAttack(entity);
+                    boolean criticalAttack = playerAttackEntityEvent.isForceCrit() || (fullStrengthAttack && this.canCriticalAttack(entity)); // Slice - force crit
                     criticalAttack = criticalAttack && !this.level().paperConfig().entities.behavior.disablePlayerCrits; // Paper - Toggleable player crits
                     if (criticalAttack) {
                         damageSource = damageSource.critical(); // Paper - critical damage API
