From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 11:07:32 -0600
Subject: [PATCH] Allow opening covered chests


diff --git a/net/minecraft/world/level/block/ChestBlock.java b/net/minecraft/world/level/block/ChestBlock.java
index 4bd84b80199c7f625475ca62da02266d871a8639..80d140188fbcfd22055362f085ed696cc974cc4c 100644
--- a/net/minecraft/world/level/block/ChestBlock.java
+++ b/net/minecraft/world/level/block/ChestBlock.java
@@ -348,8 +348,7 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
     }
 
     private static boolean isBlockedChestByBlock(final BlockGetter level, final BlockPos pos) {
-        BlockPos above = pos.above();
-        return level.getBlockState(above).isRedstoneConductor(level, above);
+        return false; // Slice - Allow opening covered chests
     }
 
     private static boolean isCatSittingOnChest(final LevelAccessor level, final BlockPos pos) {
diff --git a/net/minecraft/world/level/block/EnderChestBlock.java b/net/minecraft/world/level/block/EnderChestBlock.java
index f1a5b9670dd04b800a8d755f32364458a16d924a..b54860bc562a1963dedaef7572928e2a92a3127e 100644
--- a/net/minecraft/world/level/block/EnderChestBlock.java
+++ b/net/minecraft/world/level/block/EnderChestBlock.java
@@ -83,7 +83,7 @@ public class EnderChestBlock extends AbstractChestBlock<EnderChestBlockEntity> i
         PlayerEnderChestContainer container = player.getEnderChestInventory();
         if (container != null && level.getBlockEntity(pos) instanceof EnderChestBlockEntity enderChest) {
             BlockPos above = pos.above();
-            if (level.getBlockState(above).isRedstoneConductor(level, above)) { // Paper - diff on change; make sure that EnderChest#isBlocked uses the same logic
+            if (false) { // level.getBlockState(above).isRedstoneConductor(level, above)) { // Slice - Allow opening covered chests // Paper - diff on change; make sure that EnderChest#isBlocked uses the same logic
                 return InteractionResult.SUCCESS;
             }
 
