From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 12 Jan 2025 09:09:58 -0600
Subject: [PATCH] Allow opening covered chests


diff --git a/net/minecraft/world/level/block/ChestBlock.java b/net/minecraft/world/level/block/ChestBlock.java
index e1c52964cf165320e5a745699d8c2dbc331fcffc..a10b65a73763baaeccd68b5b364085adb8c430c6 100644
--- a/net/minecraft/world/level/block/ChestBlock.java
+++ b/net/minecraft/world/level/block/ChestBlock.java
@@ -357,8 +357,7 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
     }
 
     public static boolean isBlockedChestByBlock(BlockGetter level, BlockPos pos) {
-        BlockPos blockPos = pos.above();
-        return level.getBlockState(blockPos).isRedstoneConductor(level, blockPos);
+        return false;
     }
 
     private static boolean isCatSittingOnChest(LevelAccessor level, BlockPos pos) {
diff --git a/net/minecraft/world/level/block/EnderChestBlock.java b/net/minecraft/world/level/block/EnderChestBlock.java
index f5533960708bdbaf2eacefbc7c7c3123b7d26502..f760b2aed127cfe5e3d9762db794e09aea798172 100644
--- a/net/minecraft/world/level/block/EnderChestBlock.java
+++ b/net/minecraft/world/level/block/EnderChestBlock.java
@@ -78,7 +78,7 @@ public class EnderChestBlock extends AbstractChestBlock<EnderChestBlockEntity> i
         PlayerEnderChestContainer enderChestInventory = player.getEnderChestInventory();
         if (enderChestInventory != null && level.getBlockEntity(pos) instanceof EnderChestBlockEntity enderChestBlockEntity) {
             BlockPos blockPos = pos.above();
-            if (level.getBlockState(blockPos).isRedstoneConductor(level, blockPos)) { // Paper - diff on change; make sure that EnderChest#isBlocked uses the same logic
+            if (false) { // level.getBlockState(blockPos).isRedstoneConductor(level, blockPos)) { // Paper - diff on change; make sure that EnderChest#isBlocked uses the same logic
                 return InteractionResult.SUCCESS;
             } else {
                 // Paper start - Fix InventoryOpenEvent cancellation - moved up;
