From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 11:54:32 -0600
Subject: [PATCH] Add PlayerData Events


diff --git a/net/minecraft/stats/ServerStatsCounter.java b/net/minecraft/stats/ServerStatsCounter.java
index f5beb001b938072e78c0ead7e0ad0499784ed1cf..e9212ccc682241c50dc59e7589147c6579f2719e 100644
--- a/net/minecraft/stats/ServerStatsCounter.java
+++ b/net/minecraft/stats/ServerStatsCounter.java
@@ -94,7 +94,7 @@ public class ServerStatsCounter extends StatsCounter {
             FileUtil.createDirectoriesSafe(this.file.getParent());
 
             try (Writer writer = Files.newBufferedWriter(this.file, StandardCharsets.UTF_8)) {
-                GSON.toJson(this.toJson(), GSON.newJsonWriter(writer));
+                GSON.toJson(this.serialize(), GSON.newJsonWriter(writer));
             }
         } catch (IOException | JsonIOException e) {
             LOGGER.error("Couldn't save stats to {}", this.file, e);
@@ -126,11 +126,16 @@ public class ServerStatsCounter extends StatsCounter {
             );
     }
 
-    protected JsonElement toJson() {
+    // Slice start
+    public String toJson() {
+        return serialize().toString();
+    }
+
+    public JsonObject serialize() { // Slice end
         JsonObject result = new JsonObject();
         result.add("stats", STATS_CODEC.encodeStart(JsonOps.INSTANCE, this.stats).getOrThrow());
         result.addProperty("DataVersion", SharedConstants.getCurrentVersion().dataVersion().version());
-        return result;
+        return result; // Slice
     }
 
     public void markAllDirty() {
diff --git a/net/minecraft/world/level/storage/PlayerDataStorage.java b/net/minecraft/world/level/storage/PlayerDataStorage.java
index eddb6ebe3fa50a818e54079b73ba117be0abd195..ca1a1fe63977d8586cd7142ab8c4793ee81ef95e 100644
--- a/net/minecraft/world/level/storage/PlayerDataStorage.java
+++ b/net/minecraft/world/level/storage/PlayerDataStorage.java
@@ -32,6 +32,7 @@ public class PlayerDataStorage {
 
     public void save(final Player player) {
         if (org.spigotmc.SpigotConfig.disablePlayerDataSaving) return; // Spigot
+        if (!new com.destroystokyo.paper.event.player.PlayerSaveDataEvent((org.bukkit.entity.Player) player.getBukkitEntity()).callEvent()) return; // Slice
         try (ProblemReporter.ScopedCollector reporter = new ProblemReporter.ScopedCollector(player.problemPath(), LOGGER)) {
             TagValueOutput output = TagValueOutput.createWithContext(reporter, player.registryAccess());
             player.saveWithoutId(output);
@@ -91,6 +92,19 @@ public class PlayerDataStorage {
     }
 
     public Optional<CompoundTag> load(final NameAndId nameAndId) {
+        // Slice start - If event supplies playerdata, use it. Otherwise just load from disk as usual
+        com.destroystokyo.paper.event.player.PlayerLoadDataEvent event = new com.destroystokyo.paper.event.player.PlayerLoadDataEvent(nameAndId.id());
+        org.bukkit.Bukkit.getPluginManager().callEvent(event);
+
+        Object playerData = event.getPlayerData();
+        if (playerData instanceof CompoundTag tag) {
+            return Optional.of(ca.spottedleaf.dataconverter.minecraft.MCDataConverter.convertTag(ca.spottedleaf.dataconverter.minecraft.datatypes.MCTypeRegistry.PLAYER,
+                    tag,
+                    NbtUtils.getDataVersion(tag),
+                    ca.spottedleaf.dataconverter.minecraft.util.Version.getCurrentVersion()));
+        }
+        // Slice end
+
         Optional<CompoundTag> optTag = this.load(nameAndId, ".dat");
         if (optTag.isEmpty()) {
             this.backup(nameAndId, ".dat");
