From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 19 Jan 2025 08:49:12 -0600
Subject: [PATCH] Add DimensionDataStorageEvents


diff --git a/net/minecraft/world/level/storage/DimensionDataStorage.java b/net/minecraft/world/level/storage/DimensionDataStorage.java
index d9a3b5a2e6495b7e22c114506c2bd1e406f58f8f..ee8dd56b463aeaf7547deb7c22390443ab12b7c2 100644
--- a/net/minecraft/world/level/storage/DimensionDataStorage.java
+++ b/net/minecraft/world/level/storage/DimensionDataStorage.java
@@ -133,7 +133,7 @@ public class DimensionDataStorage implements AutoCloseable {
     }
 
     public CompletableFuture<?> scheduleSave() {
-        Map<Path, CompoundTag> map = this.collectDirtyTagsToSave();
+        Map<String, CompoundTag> map = this.collectDirtyTagsToSave();
         if (map.isEmpty()) {
             return CompletableFuture.completedFuture(null);
         } else {
@@ -144,10 +144,10 @@ public class DimensionDataStorage implements AutoCloseable {
                     List<CompletableFuture<?>> list = new ArrayList<>(i);
                     int i1 = Mth.positiveCeilDiv(size, i);
 
-                    for (List<Entry<Path, CompoundTag>> list1 : Iterables.partition(map.entrySet(), i1)) {
+                    for (List<Entry<String, CompoundTag>> list1 : Iterables.partition(map.entrySet(), i1)) {
                         list.add(CompletableFuture.runAsync(() -> {
-                            for (Entry<Path, CompoundTag> entry : list1) {
-                                tryWrite(entry.getKey(), entry.getValue());
+                            for (Entry<String, CompoundTag> entry : list1) {
+                                tryWrite(entry.getKey(), this.getDataFile(entry.getKey()), entry.getValue());
                             }
                         }, Util.ioPool()));
                     }
@@ -160,7 +160,7 @@ public class DimensionDataStorage implements AutoCloseable {
                         object -> CompletableFuture.allOf(
                             map.entrySet()
                                 .stream()
-                                .map(entry -> CompletableFuture.runAsync(() -> tryWrite(entry.getKey(), entry.getValue()), Util.DIMENSION_DATA_IO_POOL)) // Paper - Separate dimension data IO pool
+                                .map(entry -> CompletableFuture.runAsync(() -> tryWrite(entry.getKey(), this.getDataFile(entry.getKey()), entry.getValue()), Util.DIMENSION_DATA_IO_POOL)) // Paper - Separate dimension data IO pool
                                 .toArray(CompletableFuture[]::new)
                         )
                     );
@@ -170,18 +170,23 @@ public class DimensionDataStorage implements AutoCloseable {
         }
     }
 
-    private Map<Path, CompoundTag> collectDirtyTagsToSave() {
-        Map<Path, CompoundTag> map = new Object2ObjectArrayMap<>();
+    private Map<String, CompoundTag> collectDirtyTagsToSave() {
+        Map<String, CompoundTag> map = new Object2ObjectArrayMap<>();
         this.cache
             .forEach(
-                (string, optional) -> optional.filter(SavedData::isDirty)
-                    .ifPresent(savedData -> map.put(this.getDataFile(string), savedData.save(this.registries)))
+                (id, optional) -> optional.filter(SavedData::isDirty)
+                    .ifPresent(savedData -> map.put(id, savedData.save(this.registries)))
             );
         return map;
     }
 
-    private static void tryWrite(Path path, CompoundTag tag) {
+    private static void tryWrite(String id, Path path, CompoundTag tag) {
         try {
+            java.io.ByteArrayOutputStream buf = new java.io.ByteArrayOutputStream();
+            NbtIo.writeCompressed(tag, buf);
+            io.papermc.paper.event.server.DimensionDataStorageSaveEvent event = new io.papermc.paper.event.server.DimensionDataStorageSaveEvent(id, buf);
+            if (!event.callEvent()) return;
+
             NbtIo.writeCompressed(tag, path);
         } catch (IOException var3) {
             LOGGER.error("Could not save data to {}", path.getFileName(), var3);
