From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 12:02:27 -0600
Subject: [PATCH] Unsaveable entities


diff --git a/net/minecraft/server/level/ServerChunkCache.java b/net/minecraft/server/level/ServerChunkCache.java
index 075e294333115738b25b64a390df323276cf39f9..9115b68139e343f531314f14008dfae515a49c83 100644
--- a/net/minecraft/server/level/ServerChunkCache.java
+++ b/net/minecraft/server/level/ServerChunkCache.java
@@ -564,6 +564,7 @@ public class ServerChunkCache extends ChunkSource implements ca.spottedleaf.moon
         if (doMobSpawning && (this.spawnEnemies || this.spawnFriendlies)) { // Paper
             // Paper start - PlayerNaturallySpawnCreaturesEvent
             for (ServerPlayer player : this.level.players()) {
+                if (player.spawnsOwnMobs) continue; // Slice
                 int chunkRange = Math.min(level.spigotConfig.mobSpawnRange, player.getBukkitEntity().getViewDistance());
                 chunkRange = Math.min(chunkRange, 8);
                 player.playerNaturallySpawnedEvent = new com.destroystokyo.paper.event.entity.PlayerNaturallySpawnCreaturesEvent(player.getBukkitEntity(), (byte) chunkRange);
diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 92884bd6f60cb2938c62194623544c8d3ca2a57a..ce3544f030c7699b0a5efa8caf8ae32ac13064b8 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -397,6 +397,7 @@ public abstract class Entity
     public long activatedTick = Integer.MIN_VALUE;
     public boolean isTemporarilyActive;
     public long activatedImmunityTick = Integer.MIN_VALUE;
+    public boolean saveable = true; // Slice
 
     public void inactiveTick() {
     }
@@ -5246,6 +5247,7 @@ public abstract class Entity
 
     @Override
     public boolean shouldBeSaved() {
+        if (!this.saveable) return false; // Slice
         return (this.removalReason == null || this.removalReason.shouldSave())
             && !this.isPassenger()
             && (!this.isVehicle() || !((ca.spottedleaf.moonrise.patches.chunk_system.entity.ChunkSystemEntity)this).moonrise$hasAnyPlayerPassengers()); // Paper - rewrite chunk system
