From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 12:06:48 -0600
Subject: [PATCH] Add Vanish, settable inv locations


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index ce3544f030c7699b0a5efa8caf8ae32ac13064b8..71403ea6e334c46e62cdebb751f86172f2facaa0 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -398,6 +398,7 @@ public abstract class Entity
     public boolean isTemporarilyActive;
     public long activatedImmunityTick = Integer.MIN_VALUE;
     public boolean saveable = true; // Slice
+    public boolean vanished; // Slice
 
     public void inactiveTick() {
     }
@@ -1473,7 +1474,7 @@ public abstract class Entity
     private boolean vibrationAndSoundEffectsFromBlock(
         final BlockPos pos, final BlockState blockState, final boolean shouldSound, final boolean shouldVibrate, final Vec3 clippedMovement
     ) {
-        if (blockState.isAir()) {
+        if (vanished || blockState.isAir()) { // Slice
             return false;
         }
 
