From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 14 Jan 2026 12:06:48 -0600
Subject: [PATCH] Add Vanish, settable inv locations


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 88ce236b5ee31d556da04965527194f819196024..b0a5ccbcb8eb71a098878e541e138fc98ceb1e08 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -384,6 +384,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     public boolean isTemporarilyActive;
     public long activatedImmunityTick = Integer.MIN_VALUE;
     public boolean saveable = true; // Slice
+    public boolean vanished; // Slice
 
     public void inactiveTick() {
     }
@@ -1382,7 +1383,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     }
 
     private boolean vibrationAndSoundEffectsFromBlock(BlockPos pos, BlockState state, boolean playStepSound, boolean broadcastGameEvent, Vec3 entityPos) {
-        if (state.isAir()) {
+        if (vanished || state.isAir()) { // Slice
             return false;
         } else {
             boolean isStateClimbable = this.isStateClimbable(state);
