From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:13:04 -0600
Subject: [PATCH] Add Player to SpongeAbsorbEvent


diff --git a/net/minecraft/world/level/block/SpongeBlock.java b/net/minecraft/world/level/block/SpongeBlock.java
index 604bd261066fbbd98e1b95602187f0b6b3f63010..e0743ecaf58953994466fbf237c6df7e9f6c2700 100644
--- a/net/minecraft/world/level/block/SpongeBlock.java
+++ b/net/minecraft/world/level/block/SpongeBlock.java
@@ -94,7 +94,8 @@ public class SpongeBlock extends Block {
         if (!snapshots.isEmpty()) {
             final org.bukkit.block.Block sponge = org.bukkit.craftbukkit.block.CraftBlock.at(level, startPos);
 
-            org.bukkit.event.block.SpongeAbsorbEvent event = new org.bukkit.event.block.SpongeAbsorbEvent(sponge, (java.util.List<org.bukkit.block.BlockState>) (java.util.List) snapshots);
+            Level.PendingBlockEvent blockEvent = level.pendingPlayerBlockEvents.remove(startPos); // Paper
+            org.bukkit.event.block.SpongeAbsorbEvent event = new org.bukkit.event.block.SpongeAbsorbEvent(sponge, blockEvent != null ? (org.bukkit.entity.Player) blockEvent.player.getBukkitEntity() : null, (java.util.List<org.bukkit.block.BlockState>) (java.util.List) snapshots);
             if (!event.callEvent()) {
                 return false;
             }
