From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:22:29 -0600
Subject: [PATCH] Add MapEvents


diff --git a/net/minecraft/server/level/ServerLevel.java b/net/minecraft/server/level/ServerLevel.java
index 3593bb5308ab7c222a662d519e50000277e09bf3..be681e6b1aeed6215f6ab99daab7a105baae4b71 100644
--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -2342,6 +2342,14 @@ public class ServerLevel extends Level implements WorldGenLevel, ServerEntityGet
     }
 
     public MapId getFreeMapId() {
+        // Slice start
+        io.papermc.paper.event.server.MapGetFreeIndexEvent event = new io.papermc.paper.event.server.MapGetFreeIndexEvent();
+        event.callEvent();
+        int newIndex = event.getIndex();
+        if (newIndex != -1) {
+            return new MapId(newIndex);
+        }
+        // Slice end
         return this.getServer().getDataStorage().computeIfAbsent(MapIndex.TYPE).getNextMapId();
     }
 
