From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:24:09 -0600
Subject: [PATCH] noEntityCollisions for Entity


diff --git a/ca/spottedleaf/moonrise/patches/collisions/CollisionUtil.java b/ca/spottedleaf/moonrise/patches/collisions/CollisionUtil.java
index 15568f598b8c61cf92563b6a7f8cae526b84484e..b0c7d876e1bde362eefe6807d42361af02b3433c 100644
--- a/ca/spottedleaf/moonrise/patches/collisions/CollisionUtil.java
+++ b/ca/spottedleaf/moonrise/patches/collisions/CollisionUtil.java
@@ -2078,6 +2078,7 @@ public final class CollisionUtil {
 
     public static boolean getEntityHardCollisions(final Level world, final Entity entity, AABB aabb,
                                                   final List<AABB> into, final int collisionFlags, final Predicate<Entity> predicate) {
+        if (entity != null && entity.noEntityCollisions) return false; // Slice
         final boolean checkOnly = (collisionFlags & COLLISION_FLAG_CHECK_ONLY) != 0;
 
         boolean ret = false;
diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 71403ea6e334c46e62cdebb751f86172f2facaa0..14cae05c26ed8f1cf1fe7d2e151da316c4d78a54 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -418,6 +418,7 @@ public abstract class Entity
     private int sectionY = Integer.MIN_VALUE;
     private int sectionZ = Integer.MIN_VALUE;
     private boolean updatingSectionStatus;
+    public boolean noEntityCollisions; // Slice
 
     @Override
     public final boolean moonrise$isHardColliding() {
