From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 19 Jan 2025 09:29:53 -0600
Subject: [PATCH] Mute bee logger warn


diff --git a/net/minecraft/world/entity/EntityType.java b/net/minecraft/world/entity/EntityType.java
index 49201d6664656ebe34c84c1c84b5ea4878729062..ba0491725e9cff30276e7103bbe6909351efa5f4 100644
--- a/net/minecraft/world/entity/EntityType.java
+++ b/net/minecraft/world/entity/EntityType.java
@@ -1370,7 +1370,9 @@ public class EntityType<T extends Entity> implements FeatureElement, EntityTypeT
                 entity.load(tag);
             },
             // Paper end - Don't fire sync event during generation
-            () -> LOGGER.warn("Skipping Entity with id {}", tag.getString("id"))
+            () -> {
+//                LOGGER.warn("Skipping Entity with id {}", tag.getString("id")); // Slice - something about bees
+            }
         );
     }
 
