From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 3 Aug 2025 12:23:00 -0500
Subject: [PATCH] noEntityCollisions for Entity


diff --git a/ca/spottedleaf/moonrise/patches/collisions/CollisionUtil.java b/ca/spottedleaf/moonrise/patches/collisions/CollisionUtil.java
index 01cd38bba2deb6cf65c82b4e4ec352a2998fd339..7bbfe99cc88885da6125115a03a5127f0ead01a8 100644
--- a/ca/spottedleaf/moonrise/patches/collisions/CollisionUtil.java
+++ b/ca/spottedleaf/moonrise/patches/collisions/CollisionUtil.java
@@ -2069,6 +2069,7 @@ public final class CollisionUtil {
 
     public static boolean getEntityHardCollisions(final Level world, final Entity entity, AABB aabb,
                                                   final List<AABB> into, final int collisionFlags, final Predicate<Entity> predicate) {
+        if (entity != null && entity.noEntityCollisions) return false; // Slice
         final boolean checkOnly = (collisionFlags & COLLISION_FLAG_CHECK_ONLY) != 0;
 
         boolean ret = false;
diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 1f3ae8874ed1bac04e007110fb5fff621cb8103f..d7489427f2313838c29b1eb2cf85ce1c9ab685e3 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -401,6 +401,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
     private int sectionY = Integer.MIN_VALUE;
     private int sectionZ = Integer.MIN_VALUE;
     private boolean updatingSectionStatus;
+    public boolean noEntityCollisions; // Slice
 
     @Override
     public final boolean moonrise$isHardColliding() {
