From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:28:49 -0600
Subject: [PATCH] Add PlayerLoadStatsEvent


diff --git a/net/minecraft/server/players/PlayerList.java b/net/minecraft/server/players/PlayerList.java
index 3360cdb11fb999d211e532ed5e6096b4d5b07600..e4ba9e0ff0ceaf86d00e90e6affda11ea45929bd 100644
--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -1173,7 +1173,7 @@ public abstract class PlayerList {
     }
     public ServerStatsCounter getPlayerStats(final GameProfile gameProfile) {
             Path targetFile = this.locateStatsFile(gameProfile);
-            return new ServerStatsCounter(this.server, targetFile);
+            return new ServerStatsCounter(this.server, targetFile, gameProfile.id()); // Slice
     }
     // CraftBukkit end
 
diff --git a/net/minecraft/stats/ServerStatsCounter.java b/net/minecraft/stats/ServerStatsCounter.java
index e9212ccc682241c50dc59e7589147c6579f2719e..6a7b573fa2f2df74c235a5abef57b201d439fc08 100644
--- a/net/minecraft/stats/ServerStatsCounter.java
+++ b/net/minecraft/stats/ServerStatsCounter.java
@@ -63,9 +63,15 @@ public class ServerStatsCounter extends StatsCounter {
         return Codec.unboundedMap(statCodec, Codec.INT);
     }
 
-    public ServerStatsCounter(final MinecraftServer server, final Path file) {
+    public ServerStatsCounter(final MinecraftServer server, final Path file, java.util.UUID uuid) { // Slice
         this.file = file;
-        if (Files.isRegularFile(file)) {
+        // Slice start - If event supplies stats, use it. Otherwise just load from disk as usual
+        com.destroystokyo.paper.event.player.PlayerLoadStatsEvent event = new com.destroystokyo.paper.event.player.PlayerLoadStatsEvent(uuid);
+        org.bukkit.Bukkit.getPluginManager().callEvent(event);
+        JsonObject providedJson = event.getStatistics();
+        if (providedJson != null) {
+            parse(server.getFixerUpper(), providedJson);
+        } else if (Files.isRegularFile(file)) {
             try (Reader reader = Files.newBufferedReader(file, StandardCharsets.UTF_8)) {
                 JsonElement element = StrictJsonParser.parse(reader);
                 this.parse(server.getFixerUpper(), element);
