From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 21 Jan 2025 11:09:11 -0600
Subject: [PATCH] Add ObjectId


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 7e429f2679e54f4ab62f5bca98ecc958fe43b3ce..63c000832a5c9220da7afff9020c940c87d33567 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -182,7 +182,17 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
             // Paper start - Folia schedulers
             synchronized (this) {
                 if (this.bukkitEntity == null) {
-                    return this.bukkitEntity = org.bukkit.craftbukkit.entity.CraftEntity.getEntity(this.level.getCraftServer(), this);
+                    this.bukkitEntity = org.bukkit.craftbukkit.entity.CraftEntity.getEntity(this.level.getCraftServer(), this);
+
+                    // Slice start - resolve ObjectId
+                    if (this.bukkitEntity instanceof org.bukkit.craftbukkit.entity.CraftHumanEntity craftHumanEntity) {
+                        io.papermc.paper.event.player.PlayerResolveObjectIdEvent event = new io.papermc.paper.event.player.PlayerResolveObjectIdEvent((org.bukkit.entity.Player) bukkitEntity);
+                        event.callEvent();
+                        craftHumanEntity.setObjectId(event.getObjectId());
+                    }
+                    // Slice end
+
+                    return this.bukkitEntity;
                 }
             }
             // Paper end - Folia schedulers
