From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 21 Jan 2025 14:56:02 -0600
Subject: [PATCH] Equipment Packet Caching


diff --git a/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java b/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
index 59198e1f248c812185cbb1d4290a79ed1a1b3f11..baf3fa32047634964ade6a07ba43d7f17ed100a3 100644
--- a/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
@@ -30,6 +30,16 @@ public class ClientboundSetEquipmentPacket implements Packet<ClientGamePacketLis
         this.slots = slots;
     }
 
+    public ClientboundSetEquipmentPacket(int id, List<Pair<EquipmentSlot, ItemStack>> equipmentList, net.minecraft.world.entity.LivingEntity entity, boolean sanitize, String tag) {
+        this.entity = id;
+        this.sanitize = sanitize;
+        slots = new java.util.ArrayList<>(equipmentList.size());
+        for (Pair<EquipmentSlot, ItemStack> pair : equipmentList) {
+            EquipmentSlot slot = pair.getFirst();
+            slots.add(Pair.of(slot, entity.getOrCreateCachedEquipmentItem(tag, slot, pair.getSecond())));
+        }
+    }
+
     private ClientboundSetEquipmentPacket(RegistryFriendlyByteBuf buffer) {
         this.entity = buffer.readVarInt();
         this.slots = Lists.newArrayList();
diff --git a/net/minecraft/server/level/ServerEntity.java b/net/minecraft/server/level/ServerEntity.java
index ed976e95de37d653b3f1499a25f27bc768c1e9d5..8ddb629ee9f4bcaa796302acf07794d11d58839f 100644
--- a/net/minecraft/server/level/ServerEntity.java
+++ b/net/minecraft/server/level/ServerEntity.java
@@ -351,19 +351,7 @@ public class ServerEntity {
         }
 
         if (this.entity instanceof LivingEntity livingEntity) {
-            List<Pair<EquipmentSlot, ItemStack>> list = Lists.newArrayList();
-
-            for (EquipmentSlot equipmentSlot : EquipmentSlot.VALUES) {
-                ItemStack itemBySlot = livingEntity.getItemBySlot(equipmentSlot);
-                if (!itemBySlot.isEmpty()) {
-                    list.add(Pair.of(equipmentSlot, itemBySlot.copy()));
-                }
-            }
-
-            if (!list.isEmpty()) {
-                consumer.accept(new ClientboundSetEquipmentPacket(this.entity.getId(), list, true)); // Paper - data sanitization
-            }
-            ((LivingEntity) this.entity).detectEquipmentUpdatesPublic(); // CraftBukkit - SPIGOT-3789: sync again immediately after sending
+            livingEntity.sendEquipment(consumer, player);
         }
 
         if (!this.entity.getPassengers().isEmpty()) {
diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index 239c443ddc9bacc08a39a8ef2ab17016a2480549..ef5cd95356b9cb2cb327dc39b7bc159c202036c0 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -302,6 +302,11 @@ public abstract class LivingEntity extends Entity implements Attackable {
     public boolean silentDeath = false; // Paper - mark entity as dying silently for cancellable death event
     public net.kyori.adventure.util.TriState frictionState = net.kyori.adventure.util.TriState.NOT_SET; // Paper - Friction API
 
+    // Slice start
+    private static final java.util.Map<String, com.google.common.cache.Cache<Integer, ItemStack>> equipmentPacketCache = new java.util.HashMap<>();
+    protected final java.util.Map<String, net.minecraft.network.protocol.game.ClientboundSetEquipmentPacket> cachedEquipmentPacket = new java.util.HashMap<>();
+    // Slice end
+
     @Override
     public float getBukkitYaw() {
         return this.getYHeadRot();
@@ -3218,6 +3223,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
         if (map != null) {
             this.handleHandSwap(map);
             if (!map.isEmpty()) {
+                cachedEquipmentPacket.clear(); // Slice - Must invalidate cached equipment map if we have changes
                 this.handleEquipmentChanges(map);
             }
         }
@@ -3311,7 +3317,25 @@ public abstract class LivingEntity extends Entity implements Attackable {
                     this.lastBodyItemStack = itemStack;
             }
         });
-        ((ServerLevel)this.level()).getChunkSource().broadcast(this, new ClientboundSetEquipmentPacket(this.getId(), list, true)); // Paper - data sanitization
+
+        // Slice start
+        net.minecraft.server.level.ChunkMap.TrackedEntity entityTracker = ((ServerLevel) this.level()).getChunkSource().chunkMap.entityMap.get(getId());
+        if (entityTracker != null) {
+            ClientboundSetEquipmentPacket packet = new ClientboundSetEquipmentPacket(this.getId(), list);
+            for (net.minecraft.server.network.ServerPlayerConnection playerConnection : entityTracker.seenBy) {
+                ServerPlayer player = playerConnection.getPlayer();
+                org.bukkit.event.player.PlayerReceiveEquipmentEvent event = new org.bukkit.event.player.PlayerReceiveEquipmentEvent(player.getBukkitEntity(), getBukkitEntity());
+                level().getCraftServer().getPluginManager().callEvent(event);
+
+                String tag = event.getTag();
+                if (tag != null) {
+                    playerConnection.send(new ClientboundSetEquipmentPacket(this.getId(), list, this, false, tag));
+                } else {
+                    playerConnection.send(packet);
+                }
+            }
+        }
+        // Slice end
     }
 
     private ItemStack getLastArmorItem(EquipmentSlot slot) {
@@ -4520,4 +4544,85 @@ public abstract class LivingEntity extends Entity implements Attackable {
 
     public record Fallsounds(SoundEvent small, SoundEvent big) {
     }
+
+
+    // Slice start
+    public static void invalidateCachedEquipment(String tag) {
+        com.google.common.cache.Cache<Integer, ItemStack> removedCache = equipmentPacketCache.remove(tag);
+        if (removedCache != null) {
+            removedCache.asMap().clear();
+        }
+    }
+
+    public ItemStack getOrCreateCachedEquipmentItem(String tag, EquipmentSlot slot, ItemStack itemStack) {
+        return equipmentPacketCache.computeIfAbsent(tag,  s -> com.google.common.cache.CacheBuilder.newBuilder()
+                        .expireAfterAccess(10, java.util.concurrent.TimeUnit.MINUTES)
+                        .build())
+                .asMap()
+                .computeIfAbsent(itemStack.hashCode(), i -> {
+                    String name = slot.name();
+
+                    //How neat is this.
+                    if (name.equals("MAINHAND")) {
+                        name = "HAND";
+                    } else if (name.equals("OFFHAND")) {
+                        name = "OFF_HAND";
+                    }
+
+                    org.bukkit.event.entity.EntityEquipmentItemLookup event = new org.bukkit.event.entity.EntityEquipmentItemLookup(getBukkitEntity(), tag, org.bukkit.inventory.EquipmentSlot.valueOf(name), CraftItemStack.asBukkitCopy(itemStack));
+                    this.level().getCraftServer().getPluginManager().callEvent(event);
+                    org.bukkit.inventory.ItemStack eventItem = event.getItemStack();
+                    return CraftItemStack.asNMSCopy(eventItem);
+                });
+    }
+    // Slice end
+
+    // Slice start
+    public void sendEquipment(java.util.function.Consumer<net.minecraft.network.protocol.Packet<net.minecraft.network.protocol.game.ClientGamePacketListener>> consumer, ServerPlayer p) {
+        org.bukkit.event.player.PlayerReceiveEquipmentEvent event = new org.bukkit.event.player.PlayerReceiveEquipmentEvent(p.getBukkitEntity(), getBukkitEntity());
+        level().getCraftServer().getPluginManager().callEvent(event);
+
+        boolean sendEquipment = !event.isCancelled();
+        String tag = event.getTag();
+        if (sendEquipment && this instanceof ServerPlayer player && tag != null) {
+            ClientboundSetEquipmentPacket equipmentPacket = player.cachedEquipmentPacket.get(tag);
+            if (equipmentPacket != null) {
+                //Event says use a tag, and our tag exists; so we simply used our entire cached packet
+                consumer.accept(equipmentPacket);
+                return;
+            }
+        }
+
+        if (sendEquipment) {
+            List<Pair<EquipmentSlot, ItemStack>> list = Lists.newArrayList();
+            Iterator iterator = EquipmentSlot.VALUES.iterator();
+
+            while (iterator.hasNext()) {
+                EquipmentSlot enumitemslot = (EquipmentSlot) iterator.next();
+                ItemStack itemstack = getItemBySlot(enumitemslot);
+
+                if (!itemstack.isEmpty()) {
+                    ItemStack finalItemStack;
+                    if (tag != null) {
+                        finalItemStack = getOrCreateCachedEquipmentItem(tag, enumitemslot, itemstack);
+                    } else {
+                        finalItemStack = itemstack.copy();
+                    }
+
+                    list.add(Pair.of(enumitemslot, finalItemStack));
+                }
+            }
+
+            if (!list.isEmpty()) {
+                ClientboundSetEquipmentPacket equipmentPacket = new ClientboundSetEquipmentPacket(getId(), list, true);
+                if (tag != null) {
+                    cachedEquipmentPacket.put(tag, equipmentPacket);
+                }
+                consumer.accept(equipmentPacket);
+            }
+
+            detectEquipmentUpdatesPublic(); // CraftBukkit - SPIGOT-3789: sync again immediately after sending
+        }
+    }
+    // Slice end
 }
