From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:40:19 -0600
Subject: [PATCH] Custom Tracking Ranges


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 14cae05c26ed8f1cf1fe7d2e151da316c4d78a54..c94eee03749f4b685db53ebf1460a14b0713ca2a 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -399,6 +399,7 @@ public abstract class Entity
     public long activatedImmunityTick = Integer.MIN_VALUE;
     public boolean saveable = true; // Slice
     public boolean vanished; // Slice
+    public int trackingRange; // Slice
 
     public void inactiveTick() {
     }
