From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 21 Jan 2025 15:00:18 -0600
Subject: [PATCH] Add Vanish


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 63c000832a5c9220da7afff9020c940c87d33567..b228607516b060c036c97433dce93966d666de59 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -353,6 +353,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
     public boolean isTemporarilyActive;
     public long activatedImmunityTick = Integer.MIN_VALUE;
     public boolean saveable = true; // Slice
+    public boolean vanished; // Slice
 
     public void inactiveTick() {
     }
@@ -1313,7 +1314,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
     }
 
     private boolean vibrationAndSoundEffectsFromBlock(BlockPos pos, BlockState state, boolean playStepSound, boolean broadcastGameEvent, Vec3 entityPos) {
-        if (state.isAir()) {
+        if (vanished || state.isAir()) { // Slice
             return false;
         } else {
             boolean isStateClimbable = this.isStateClimbable(state);
