From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:50:44 -0600
Subject: [PATCH] Expose Entity.despawnTime


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index c94eee03749f4b685db53ebf1460a14b0713ca2a..237713ca7c93aaf3168eeee853ce32790790e505 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -389,7 +389,7 @@ public abstract class Entity
     // Paper end
     public boolean freezeLocked = false; // Paper - Freeze Tick Lock API
     public boolean fixedPose = false; // Paper - Expand Pose API
-    private final int despawnTime; // Paper - entity despawn time limit
+    public int despawnTime; // Paper - entity despawn time limit // Slice (public, not final)
     public int totalEntityAge; // Paper - age-like counter for all entities
     public final io.papermc.paper.entity.activation.ActivationType activationType = io.papermc.paper.entity.activation.ActivationType.activationTypeFor(this); // Paper - EAR 2/tracking ranges
     // Paper start - EAR 2
