From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:50:44 -0600
Subject: [PATCH] Expose Entity.despawnTime


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index fa4397460db1b7bb7b0c6acf4d6479089f5a8386..12288af8f5d82000ebbb2435e5d649532a6d6d40 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -375,7 +375,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     // Paper end
     public boolean freezeLocked = false; // Paper - Freeze Tick Lock API
     public boolean fixedPose = false; // Paper - Expand Pose API
-    private final int despawnTime; // Paper - entity despawn time limit
+    public int despawnTime; // Paper - entity despawn time limit // Slice (public, not final)
     public int totalEntityAge; // Paper - age-like counter for all entities
     public final io.papermc.paper.entity.activation.ActivationType activationType = io.papermc.paper.entity.activation.ActivationType.activationTypeFor(this); // Paper - EAR 2/tracking ranges
     // Paper start - EAR 2
