From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 21 Jan 2025 15:23:16 -0600
Subject: [PATCH] Add PlayerLoadStatsEvent


diff --git a/net/minecraft/server/players/PlayerList.java b/net/minecraft/server/players/PlayerList.java
index b3789a805ab7ca349b9891630b0db234f12753bd..dd82baf5c2797585811ce16b0d5af92cf43b09ec 100644
--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -1316,7 +1316,7 @@ public abstract class PlayerList {
                 }
             }
 
-            serverStatsCounter = new ServerStatsCounter(this.server, file1);
+            serverStatsCounter = new ServerStatsCounter(this.server, file1, uuid); // Slice
             // this.stats.put(uuid, serverStatsCounter); // CraftBukkit
         }
 
diff --git a/net/minecraft/stats/ServerStatsCounter.java b/net/minecraft/stats/ServerStatsCounter.java
index 743a10eb787031ce84c5c4b431b1e1d06db04259..d6eb42d75c592ba48d49bda720a61aa4f86a4c24 100644
--- a/net/minecraft/stats/ServerStatsCounter.java
+++ b/net/minecraft/stats/ServerStatsCounter.java
@@ -39,10 +39,16 @@ public class ServerStatsCounter extends StatsCounter {
     private final File file;
     private final Set<Stat<?>> dirty = Sets.newHashSet();
 
-    public ServerStatsCounter(MinecraftServer server, File file) {
+    public ServerStatsCounter(MinecraftServer server, File file, java.util.UUID uuid) { // Slice
         this.server = server;
         this.file = file;
-        if (file.isFile()) {
+        // Slice start - If event supplies stats, use it. Otherwise just load from disk as usual
+        com.destroystokyo.paper.event.player.PlayerLoadStatsEvent event = new com.destroystokyo.paper.event.player.PlayerLoadStatsEvent(uuid);
+        org.bukkit.Bukkit.getPluginManager().callEvent(event);
+        JsonObject providedJson = event.getStatistics();
+        if (providedJson != null) {
+            parseLocal(server.getFixerUpper(), providedJson);
+        } else if (file.isFile()) { // Slice end
             try {
                 this.parseLocal(server.getFixerUpper(), FileUtils.readFileToString(file));
             } catch (IOException var4) {
@@ -184,6 +190,43 @@ public class ServerStatsCounter extends StatsCounter {
         return jsonObject1; // Slice
     }
 
+    public void parseLocal(DataFixer dataFixer, JsonObject jsonElement) {
+        CompoundTag compoundTag = fromJson(jsonElement.getAsJsonObject());
+        compoundTag = DataFixTypes.STATS.updateToCurrentVersion(dataFixer, compoundTag, NbtUtils.getDataVersion(compoundTag, 1343));
+        if (!compoundTag.contains("stats", 10)) {
+            return;
+        }
+
+        CompoundTag compound = compoundTag.getCompound("stats");
+
+        for (String string : compound.getAllKeys()) {
+            if (compound.contains(string, 10)) {
+                Util.ifElse(
+                        BuiltInRegistries.STAT_TYPE.getOptional(ResourceLocation.parse(string)),
+                        type -> {
+                            CompoundTag compound1 = compound.getCompound(string);
+
+                            for (String string1 : compound1.getAllKeys()) {
+                                if (compound1.contains(string1, 99)) {
+                                    Util.ifElse(
+                                            this.getStat(type, string1),
+                                            stat -> this.stats.put(stat, compound1.getInt(string1)),
+                                            () -> LOGGER.warn("Invalid statistic in {}: Don't know what {} is", this.file, string1)
+                                    );
+                                } else {
+                                    LOGGER.warn(
+                                            "Invalid statistic value in {}: Don't know what {} is for key {}", this.file, compound1.get(string1), string1
+                                    );
+                                }
+                            }
+                        },
+                        () -> LOGGER.warn("Invalid statistic type in {}: Don't know what {} is", this.file, string)
+                );
+            }
+        }
+    }
+    // Slice end
+
     private static <T> ResourceLocation getKey(Stat<T> stat) {
         return stat.getType().getRegistry().getKey(stat.getValue());
     }
