From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:56:12 -0600
Subject: [PATCH] Add LeafBlockTickEvent


diff --git a/net/minecraft/world/level/block/LeavesBlock.java b/net/minecraft/world/level/block/LeavesBlock.java
index d5b114dca7b28d910f03ae3468e891d5d9863897..faf491ad7cd3fffe4c7caa7a385dc530ed19b1b0 100644
--- a/net/minecraft/world/level/block/LeavesBlock.java
+++ b/net/minecraft/world/level/block/LeavesBlock.java
@@ -60,14 +60,16 @@ public abstract class LeavesBlock extends Block implements SimpleWaterloggedBloc
 
     @Override
     protected boolean isRandomlyTicking(BlockState state) {
-        return state.getValue(DISTANCE) == 7 && !state.getValue(PERSISTENT);
+        return true; //state.getValue(DISTANCE) == 7 && !state.getValue(PERSISTENT);
     }
 
     @Override
     protected void randomTick(BlockState state, ServerLevel level, BlockPos pos, RandomSource random) {
+        org.bukkit.craftbukkit.block.CraftBlock block = org.bukkit.craftbukkit.block.CraftBlock.at(level, pos);
+        new io.papermc.paper.event.block.LeafBlockTickEvent(block).callEvent();
         if (this.decaying(state)) {
             // CraftBukkit start
-            org.bukkit.event.block.LeavesDecayEvent event = new org.bukkit.event.block.LeavesDecayEvent(org.bukkit.craftbukkit.block.CraftBlock.at(level, pos));
+            org.bukkit.event.block.LeavesDecayEvent event = new org.bukkit.event.block.LeavesDecayEvent(block);
             level.getCraftServer().getPluginManager().callEvent(event);
 
             if (event.isCancelled() || !level.getBlockState(pos).is(this)) {
