From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 08:57:00 -0600
Subject: [PATCH] Add EntityTouchWaterEvent


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 237713ca7c93aaf3168eeee853ce32790790e505..55d730bb55fe5fa8fbf7a4dc2aa4576b72e26f4d 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -2183,7 +2183,10 @@ public abstract class Entity
             this.resetFallDistance();
             if (!this.wasTouchingWater && !this.firstTick) {
                 this.doWaterSplashEffect();
+                new io.papermc.paper.event.entity.EntityTouchWaterEvent(getBukkitEntity(), true).callEvent(); // Slice
             }
+        } else if (this.wasTouchingWater) {
+            new io.papermc.paper.event.entity.EntityTouchWaterEvent(getBukkitEntity(), false).callEvent(); // Slice
         }
 
         this.wasTouchingWater = inWater;
