From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Mon, 27 Jan 2025 08:35:24 -0600
Subject: [PATCH] Don't send fire packets if player has FR


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 0faa3a0ef9a56d0e7d25fe0752b4dc05ce09790b..fa7d48f15e5a69b4996d86a59142a69724082ca9 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -895,7 +895,11 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
 
         this.checkBelowWorld();
         if (!this.level().isClientSide) {
-            this.setSharedFlagOnFire(this.remainingFireTicks > 0);
+            if (this instanceof net.minecraft.world.entity.LivingEntity livingEntity) {
+                this.setSharedFlagOnFire(this.remainingFireTicks > 0 && !livingEntity.hasEffect(net.minecraft.world.effect.MobEffects.FIRE_RESISTANCE));
+            } else {
+                this.setSharedFlagOnFire(this.remainingFireTicks > 0);
+            }
         }
 
         this.firstTick = false;
