From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:03:42 -0600
Subject: [PATCH] Add Player.getLastPing


diff --git a/net/minecraft/server/network/ServerCommonPacketListenerImpl.java b/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
index 5dbb2849f32ac3418f8fdf76c04022d7e9dc165b..7dc80511b9da9d881bcfe6dd3b090c55669643c7 100644
--- a/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
@@ -45,6 +45,7 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
     private long closedListenerTime;
     private boolean closed = false;
     private volatile int latency; // Paper - improve keepalives - make volatile
+    private volatile int lastLatency; // Paper
     private final io.papermc.paper.util.KeepAlive keepAlive; // Paper - improve keepalives
     private volatile boolean suspendFlushingOnServerThread = false;
     // CraftBukkit start
@@ -63,6 +64,7 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
         this.connection = connection;
         //this.keepAliveTime = Util.getMillis(); // Paper - improve keepalives
         this.latency = cookie.latency();
+        this.lastLatency = this.latency; // Paper
         this.transferred = cookie.transferred();
         // Paper start
         this.clientBrand = cookie.clientBrand();
@@ -113,6 +115,7 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
             this.keepAlive.pingCalculator5s.update(response);
 
             this.latency = this.keepAlive.pingCalculator5s.getAvgLatencyMS();
+            this.lastLatency = (int) java.util.concurrent.TimeUnit.NANOSECONDS.toMillis(response.latencyNS()); // Paper
             return;
         }
 
@@ -446,6 +449,8 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
         return this.latency;
     }
 
+    public int lastLatency() { return this.lastLatency; } // Paper
+
     protected CommonListenerCookie createCookie(final ClientInformation clientInformation) {
         // Paper start - listener handoff should reset pending keepalive expectations
         return new CommonListenerCookie(
