From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:29:24 -0600
Subject: [PATCH] Packet obfuscation and reduction


diff --git a/net/minecraft/network/syncher/SynchedEntityData.java b/net/minecraft/network/syncher/SynchedEntityData.java
index 0075164bf057271d63a8b97473aa5c41c03eac91..c42b349ec5f2a44b50bb0f6f748cb7cc05772cb5 100644
--- a/net/minecraft/network/syncher/SynchedEntityData.java
+++ b/net/minecraft/network/syncher/SynchedEntityData.java
@@ -54,11 +54,27 @@ public class SynchedEntityData {
     }
 
     public <T> void set(final EntityDataAccessor<T> accessor, final T value) {
-        this.set(accessor, value, false);
+        // Slice start
+        this.set(accessor, value, false, null);
     }
 
     public <T> void set(final EntityDataAccessor<T> accessor, final T value, final boolean forceDirty) {
+        this.set(accessor, value, forceDirty, null);
+    }
+
+    public <T> void set(final EntityDataAccessor<T> accessor, final T value, final @Nullable T foreignValue) {
+        this.set(accessor, value, false, foreignValue);
+    }
+
+    public <T> void set(final EntityDataAccessor<T> accessor, final T value, final boolean forceDirty, final @Nullable T foreignValue) { // Slice end
         SynchedEntityData.DataItem<T> dataItem = this.getItem(accessor);
+
+        // Slice start
+        if (foreignValue != null && ObjectUtils.notEqual(foreignValue, dataItem.getForeignValue())) {
+            dataItem.setForeignValue(foreignValue);
+        }
+        // Slice end
+
         if (forceDirty || ObjectUtils.notEqual(value, dataItem.getValue())) {
             dataItem.setValue(value);
             this.entity.onSyncedDataUpdated(accessor);
@@ -78,38 +94,66 @@ public class SynchedEntityData {
         return this.isDirty;
     }
 
-    public @Nullable List<SynchedEntityData.DataValue<?>> packDirty() {
+    public org.apache.commons.lang3.tuple.Pair<List<SynchedEntityData.DataValue<?>>, List<SynchedEntityData.DataValue<?>>> packDirty() {
         if (!this.isDirty) {
             return null;
         }
 
         this.isDirty = false;
-        List<SynchedEntityData.DataValue<?>> result = new ArrayList<>();
+        List<SynchedEntityData.DataValue<?>> result = new ArrayList<>(this.itemsById.length); // Slice
+        List<SynchedEntityData.DataValue<?>> foreignList = null; // Slice
 
         for (SynchedEntityData.DataItem<?> dataItem : this.itemsById) {
             if (dataItem.isDirty()) {
+                DataValue<?> value = dataItem.value(); // Slice
                 dataItem.setDirty(false);
-                result.add(dataItem.value());
+                result.add(value);
+
+                // Slice start
+                boolean foreignDirty = dataItem.isForeignDirty();
+                if (foreignDirty || dataItem.foreignValue == null) {
+                    if (foreignList == null) {
+                        foreignList = new ArrayList<>(this.itemsById.length);
+                    }
+
+                    if (foreignDirty) {
+                        dataItem.setForeignDirty(false);
+                        foreignList.add(dataItem.foreignValue != null ? dataItem.foreignValue() : dataItem.value());
+                    } else {
+                        foreignList.add(dataItem.value());
+                    }
+                }
+                // Slice end
             }
         }
 
-        return result;
+        return org.apache.commons.lang3.tuple.Pair.of(result, foreignList); // Slice
     }
 
-    public @Nullable List<SynchedEntityData.DataValue<?>> getNonDefaultValues() {
+    public org.apache.commons.lang3.tuple.Pair<List<SynchedEntityData.DataValue<?>>, List<SynchedEntityData.DataValue<?>>> getNonDefaultValues() {
         List<SynchedEntityData.DataValue<?>> result = null;
+        List<SynchedEntityData.DataValue<?>> foreignList = null; // Slice
 
         for (SynchedEntityData.DataItem<?> dataItem : this.itemsById) {
             if (!dataItem.isSetToDefault()) {
                 if (result == null) {
-                    result = new ArrayList<>();
+                    result = new ArrayList<>(this.itemsById.length);
                 }
 
                 result.add(dataItem.value());
+                // Slice start
+                if (dataItem.foreignValue == null || !dataItem.foreignValue.equals(dataItem.initialValue)) {
+                    if (foreignList == null) {
+                        foreignList = new ArrayList<>(this.itemsById.length);
+                    }
+
+                    foreignList.add(dataItem.foreignValue != null ? dataItem.foreignValue() : dataItem.value());
+                }
+                // Slice end
             }
         }
 
-        return result;
+        return org.apache.commons.lang3.tuple.Pair.of(result, foreignList); // Slice
     }
 
     public void assignValues(final List<SynchedEntityData.DataValue<?>> items) {
@@ -197,11 +241,14 @@ public class SynchedEntityData {
         private T value;
         private final T initialValue;
         private boolean dirty;
+        private @Nullable T foreignValue = null; // Slice
+        private boolean foreignDirty; // Slice
 
         public DataItem(final EntityDataAccessor<T> accessor, final T initialValue) {
             this.accessor = accessor;
             this.initialValue = initialValue;
             this.value = initialValue;
+            this.foreignDirty = true; //Slice
         }
 
         public EntityDataAccessor<T> getAccessor() {
@@ -231,6 +278,32 @@ public class SynchedEntityData {
         public SynchedEntityData.DataValue<T> value() {
             return SynchedEntityData.DataValue.create(this.accessor, this.value);
         }
+
+
+        // Slice start
+        public SynchedEntityData.DataValue<T> foreignValue() {
+            return SynchedEntityData.DataValue.create(this.accessor, this.foreignValue);
+        }
+
+        public void setForeignValue(T foreignValue) {
+            this.foreignValue = foreignValue;
+            this.foreignDirty = true;
+        }
+
+        public @Nullable T getForeignValue() {
+            return foreignValue;
+        }
+
+        public boolean isForeignDirty() {
+            //There must be a foreign value in order for this to be dirty, otherwise we consider this a normal
+            //value and check the normal dirty flag.
+            return foreignValue == null || this.foreignDirty;
+        }
+
+        public void setForeignDirty(boolean dirty) {
+            this.foreignDirty = dirty;
+        }
+        // Slice end
     }
 
     public record DataValue<T>(int id, EntityDataSerializer<T> serializer, T value) {
diff --git a/net/minecraft/server/level/ServerEntity.java b/net/minecraft/server/level/ServerEntity.java
index e9f54ee0d58205b8116dd63bfc18d38325650b04..fb5e1fbe52cc75fda88c4ff16d894843476a3d46 100644
--- a/net/minecraft/server/level/ServerEntity.java
+++ b/net/minecraft/server/level/ServerEntity.java
@@ -69,6 +69,7 @@ public class ServerEntity {
     private boolean wasRiding;
     private boolean wasOnGround;
     private @Nullable List<SynchedEntityData.DataValue<?>> trackedDataValues;
+    private List<SynchedEntityData.DataValue<?>> trackedForeignDataValues; // Slice
     private final Set<net.minecraft.server.network.ServerPlayerConnection> trackedPlayers; // Paper
 
     public ServerEntity(
@@ -86,7 +87,9 @@ public class ServerEntity {
         this.lastSentXRot = Mth.packDegrees(entity.getXRot());
         this.lastSentYHeadRot = Mth.packDegrees(entity.getYHeadRot());
         this.wasOnGround = entity.onGround();
-        this.trackedDataValues = entity.getEntityData().getNonDefaultValues();
+        org.apache.commons.lang3.tuple.Pair<List<SynchedEntityData.DataValue<?>>, List<SynchedEntityData.DataValue<?>>> nonDefaultValues = entity.getEntityData().getNonDefaultValues();
+        this.trackedDataValues = nonDefaultValues.getLeft();
+        this.trackedForeignDataValues = nonDefaultValues.getRight();
     }
 
     // Paper start - fix desync when a player is added to the tracker
@@ -328,20 +331,31 @@ public class ServerEntity {
 
         Packet<ClientGamePacketListener> packet = this.entity.getAddEntityPacket(this);
         broadcast.accept(packet);
-        if (this.trackedDataValues != null) {
-            broadcast.accept(new ClientboundSetEntityDataPacket(this.entity.getId(), this.trackedDataValues));
+        List<SynchedEntityData.DataValue<?>> dataValues = this.entity instanceof ServerPlayer ? this.trackedForeignDataValues : this.trackedDataValues; // Slice
+        if (dataValues != null) {
+            broadcast.accept(new ClientboundSetEntityDataPacket(this.entity.getId(), dataValues));
         }
 
-        if (this.entity instanceof LivingEntity livingEntity) {
+        if (this.entity instanceof LivingEntity livingEntity) { // Slice start - only do this if this entity broadcasts attributes to others
             Collection<AttributeInstance> attributes = livingEntity.getAttributes().getSyncableAttributes();
-            if (!attributes.isEmpty()) {
-                // CraftBukkit start - If sending own attributes send scaled health instead of current maximum health
-                if (this.entity.getId() == player.getId()) {
-                    ((ServerPlayer) this.entity).getBukkitEntity().injectScaledMaxHealth(attributes, false);
+            if (livingEntity.broadcastsAttributes) {
+                if (!attributes.isEmpty()) {
+                    // CraftBukkit start - If sending own attributes send scaled health instead of current maximum health
+                    if (this.entity.getId() == player.getId()) {
+                        ((ServerPlayer) this.entity).getBukkitEntity().injectScaledMaxHealth(attributes, false);
+                    }
+                    // CraftBukkit end
+                    broadcast.accept(new ClientboundUpdateAttributesPacket(this.entity.getId(), attributes));
                 }
-                // CraftBukkit end
-                broadcast.accept(new ClientboundUpdateAttributesPacket(this.entity.getId(), attributes));
-            }
+            } else {
+                //Always send the scale key if it's one of them.
+                for (AttributeInstance instance : attributes) {
+                    if (instance.getAttribute().is(SCALE_ATTRIBUTE_KEY)) {
+                        broadcast.accept(new ClientboundUpdateAttributesPacket(this.entity.getId(), List.of(instance)));
+                        break;
+                    }
+                }
+            } // Slice end
         }
 
         if (this.entity instanceof LivingEntity livingEntity) {
@@ -381,23 +395,58 @@ public class ServerEntity {
         return Mth.unpackDegrees(this.lastSentYHeadRot);
     }
 
+    private static final net.minecraft.resources.Identifier SCALE_ATTRIBUTE_KEY = org.bukkit.craftbukkit.util.CraftNamespacedKey.toMinecraft(org.bukkit.attribute.Attribute.SCALE.getKey()); // Slice
     private void sendDirtyEntityData() {
         SynchedEntityData entityData = this.entity.getEntityData();
-        List<SynchedEntityData.DataValue<?>> packedValues = entityData.packDirty();
+        org.apache.commons.lang3.tuple.Pair<List<SynchedEntityData.DataValue<?>>, List<SynchedEntityData.DataValue<?>>> packedValues = entityData.packDirty(); // Slice
         if (packedValues != null) {
-            this.trackedDataValues = entityData.getNonDefaultValues();
-            this.synchronizer.sendToTrackingPlayersAndSelf(new ClientboundSetEntityDataPacket(this.entity.getId(), packedValues));
+            // Slice start
+            org.apache.commons.lang3.tuple.Pair<List<SynchedEntityData.DataValue<?>>, List<SynchedEntityData.DataValue<?>>> nonDefaultValues = entityData.getNonDefaultValues();
+            this.trackedDataValues = nonDefaultValues.getLeft();
+            this.trackedForeignDataValues = nonDefaultValues.getRight();
+
+            //We always send any dirty values to ourself.
+            List<SynchedEntityData.DataValue<?>> dirtyValues = packedValues.getLeft();
+            if (this.entity instanceof ServerPlayer serverPlayer) {
+                serverPlayer.connection.send(new ClientboundSetEntityDataPacket(this.entity.getId(), dirtyValues));
+            }
+
+            //But only send dirty foreign values to other clients.
+            List<SynchedEntityData.DataValue<?>> foreignDirtyValues = packedValues.getRight();
+            if (foreignDirtyValues != null) {
+                this.synchronizer.sendToTrackingPlayersAndSelf(new ClientboundSetEntityDataPacket(this.entity.getId(), foreignDirtyValues));
+            }
+            // Slice end
         }
 
         if (this.entity instanceof LivingEntity livingEntity) {
             Set<AttributeInstance> attributes = livingEntity.getAttributes().getAttributesToSync();
             if (!attributes.isEmpty()) {
+                boolean broadcastAttributes = livingEntity.broadcastsAttributes; // Slice
                 // CraftBukkit start - Send scaled max health
                 if (this.entity instanceof ServerPlayer serverPlayer) {
                     serverPlayer.getBukkitEntity().injectScaledMaxHealth(attributes, false);
+
+                    // Slice - If not broadcasting, we still need to send it to ourself.
+                    if (!broadcastAttributes) {
+                        serverPlayer.connection.send(new ClientboundUpdateAttributesPacket(this.entity.getId(), attributes));
+                    }
                 }
                 // CraftBukkit end
-                this.synchronizer.sendToTrackingPlayersAndSelf(new ClientboundUpdateAttributesPacket(this.entity.getId(), attributes));
+
+                // Slice start
+                if (broadcastAttributes) {
+                    this.synchronizer.sendToTrackingPlayersAndSelf(new ClientboundUpdateAttributesPacket(this.entity.getId(), attributes));
+                } else {
+                    //Always send the scale key if it's one of them.
+                    for (AttributeInstance instance : attributes) {
+                        if (instance.getAttribute().is(SCALE_ATTRIBUTE_KEY)) {
+                            this.synchronizer.sendToTrackingPlayersAndSelf(new ClientboundUpdateAttributesPacket(this.entity.getId(), List.of(instance)));
+                            break;
+                        }
+                    }
+                }
+                // Slice end
             }
 
             attributes.clear();
diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 55d730bb55fe5fa8fbf7a4dc2aa4576b72e26f4d..3c0fb43cfed1eaa73f9541f68e0f73c74a1b28ff 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -3811,7 +3811,7 @@ public abstract class Entity
             }
             return;
         }
-        this.entityData.set(DATA_AIR_SUPPLY_ID, event.getAmount());
+        this.entityData.set(DATA_AIR_SUPPLY_ID, event.getAmount(), getMaxAirSupply()); // Slice
         // CraftBukkit end
     }
 
diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index 585a781a565936a5ab153d095de5570e1883dd85..a4e743925d2e9c5b3771033097cc582b6edd3e13 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -307,6 +307,9 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
     public int invulnerableDuration = LivingEntity.INVULNERABLE_DURATION; // Paper - configurable invulnerable duration
     // CraftBukkit end
 
+    public boolean broadcastsPotionEffects = true; // Slice
+    public boolean broadcastsAttributes = true; // Slice
+
     protected LivingEntity(final EntityType<? extends LivingEntity> type, final Level level) {
         super(type, level);
         this.attributes = new AttributeMap(DefaultAttributes.getSupplier(type));
@@ -319,6 +322,7 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
         this.setYRot(this.random.nextFloat() * (float) (Math.PI * 2));
         this.yHeadRot = this.getYRot();
         this.brain = this.makeBrain(Brain.Packed.EMPTY);
+        if (this instanceof ServerPlayer) broadcastsAttributes = false; // Slice
     }
 
     @Override
@@ -985,6 +989,8 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
                 effect.tickClient();
             }
 
+            if (!broadcastsPotionEffects) return; // Slice - Simply don't emit ambient particle fx if this is off
+
             List<ParticleOptions> particles = this.entityData.get(DATA_EFFECT_PARTICLES);
             if (!particles.isEmpty()) {
                 boolean isAmbient = this.entityData.get(DATA_EFFECT_AMBIENCE_ID);
@@ -1017,6 +1023,13 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
     }
 
     private void updateSynchronizedMobEffectParticles() {
+        // Slice start
+        if (!broadcastsPotionEffects) {
+            updateSynchronizedMobEffectParticlesNoBroadcast();
+            return;
+        }
+        // Slice end
+
         List<ParticleOptions> visibleEffectParticles = this.activeEffects
             .values()
             .stream()
@@ -1027,6 +1040,38 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
         this.entityData.set(DATA_EFFECT_AMBIENCE_ID, areAllEffectsAmbient(this.activeEffects.values()));
     }
 
+    // Slice Start - When an entity broadcasts no particles, we want to still broadcast Invis ambient particles
+    // so players can still see those and track invis players at least.
+    private void updateSynchronizedMobEffectParticlesNoBroadcast() {
+        boolean allEffectsAmbient = true;
+        List<ParticleOptions> particleOptionsLocal = new java.util.ArrayList<>(this.activeEffects.size());
+        List<ParticleOptions> particleOptionsForeign = null;
+        for (Entry<Holder<MobEffect>, MobEffectInstance> entry : this.activeEffects.entrySet()) {
+            MobEffectInstance effect = entry.getValue();
+            boolean visible = effect.isVisible();
+            if (visible && !effect.isAmbient() && allEffectsAmbient) {
+                allEffectsAmbient = false;
+            }
+
+            if (visible) {
+                ParticleOptions options = effect.getParticleOptions();
+                particleOptionsLocal.add(options);
+
+                if (entry.getKey().equals(org.bukkit.craftbukkit.potion.CraftPotionEffectType.bukkitToMinecraftHolder(org.bukkit.potion.PotionEffectType.INVISIBILITY))) {
+                    if (particleOptionsForeign == null) {
+                        particleOptionsForeign = new java.util.ArrayList<>(this.activeEffects.size());
+                    }
+
+                    particleOptionsForeign.add(options);
+                }
+            }
+        }
+
+        this.entityData.set(DATA_EFFECT_PARTICLES, particleOptionsLocal, particleOptionsForeign != null ? particleOptionsForeign : java.util.Collections.emptyList());
+        this.entityData.set(DATA_EFFECT_AMBIENCE_ID, allEffectsAmbient, false, false);
+    }
+    // Slice end
+
     private void updateGlowingStatus() {
         boolean glowingState = this.isCurrentlyGlowing();
         if (this.getSharedFlag(Entity.FLAG_GLOWING) != glowingState) {
diff --git a/net/minecraft/world/entity/ai/attributes/AttributeMap.java b/net/minecraft/world/entity/ai/attributes/AttributeMap.java
index 68bad6752bd313ca3ebde7fd4e3b2c46c5dee3ef..96f45d1e1fb09e9f794ab0bcae5ea8efb85b2e1f 100644
--- a/net/minecraft/world/entity/ai/attributes/AttributeMap.java
+++ b/net/minecraft/world/entity/ai/attributes/AttributeMap.java
@@ -39,7 +39,15 @@ public class AttributeMap {
     }
 
     public Collection<AttributeInstance> getSyncableAttributes() {
-        return this.attributes.values().stream().filter(instance -> instance.getAttribute().value().isClientSyncable()).collect(Collectors.toList());
+        // Slice start - Undo stream here, this is fairly hot.
+        java.util.List<AttributeInstance> list = new java.util.ArrayList<>(this.attributes.size());
+        for (AttributeInstance instance : this.attributes.values()) {
+            if (instance.getAttribute().value().isClientSyncable()) {
+                list.add(instance);
+            }
+        }
+        return list;
+        // Slice end
     }
 
     public @Nullable AttributeInstance getInstance(final Holder<Attribute> attribute) {
diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index 742e6d2d3ae7d514cc32d71c2ae026d40f4b8cf2..4f68ddc1da21dc665619df3d404396f35ab68bd6 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -543,7 +543,7 @@ public abstract class Player extends Avatar implements ContainerUser {
 
     public void increaseScore(final int amount) {
         int score = this.getScore();
-        this.entityData.set(DATA_SCORE_ID, score + amount);
+        this.entityData.set(DATA_SCORE_ID, score + amount, 0); // Slice
     }
 
     public void startAutoSpinAttack(final int activationTicks, final float dmg, final ItemStack itemStackUsed) {
