From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:37:33 -0600
Subject: [PATCH] Allow inventory clicks in Spectator


diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index f936da2a1ffef5fd84d8d63be3575fd06ac9a7b2..19a457003eb62f448539783e71ce1faf57d03edb 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3053,7 +3053,7 @@ public class ServerGamePacketListenerImpl
         if (this.player.isImmobile()) return; // CraftBukkit
         this.player.resetLastActionTime();
         if (this.player.containerMenu.containerId == packet.containerId() && this.player.containerMenu.stillValid(this.player)) { // CraftBukkit
-            boolean cancelled = this.player.isSpectator(); // CraftBukkit - pre cancelled when in spectator
+            boolean cancelled = false; //Slice - allow spectator inv management this.player.isSpectator(); // CraftBukkit - pre cancelled when in spectator
             if ((false && this.player.isSpectator()) || this.player.isDeadOrDying()) { // CraftBukkit
                 this.player.containerMenu.sendAllDataToRemote();
             } else if (!this.player.containerMenu.stillValid(this.player)) {
