From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:37:52 -0600
Subject: [PATCH] Don't send fire packets if player not on fire


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index ac1fedeb0119c52dc278324c8cdbf83fb6d15d8e..b6cddcabc2c464fe3cf9a4f120dbb732cab21008 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -914,7 +914,11 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
 
         this.checkBelowWorld();
         if (!this.level().isClientSide()) {
-            this.setSharedFlagOnFire(this.remainingFireTicks > 0);
+            if (this instanceof net.minecraft.world.entity.LivingEntity livingEntity) {
+                this.setSharedFlagOnFire(this.remainingFireTicks > 0 && !livingEntity.hasEffect(net.minecraft.world.effect.MobEffects.FIRE_RESISTANCE));
+            } else {
+                this.setSharedFlagOnFire(this.remainingFireTicks > 0);
+            }
         }
 
         this.firstTick = false;
