From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:38:05 -0600
Subject: [PATCH] Long-range tracking for spectators


diff --git a/net/minecraft/server/level/ChunkMap.java b/net/minecraft/server/level/ChunkMap.java
index ab8893163ac85971a470e4a4d6fa76bd174e2bae..2b9c838fc8e0f65f03b38c06bbed9cd3cb1a8261 100644
--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -1363,7 +1363,7 @@ public class ChunkMap extends SimpleRegionStorage implements ChunkHolder.PlayerP
                 double vec3_dz = player.getZ() - this.entity.getZ();
                 // Paper end - remove allocation of Vec3D here
                 int playerViewDistance = ChunkMap.this.getPlayerViewDistance(player);
-                double d = Math.min(this.getEffectiveRange(), playerViewDistance * 16);
+                double d = player.gameMode.getGameModeForPlayer() == net.minecraft.world.level.GameType.SPECTATOR ? 128 : Math.min(this.getEffectiveRange(), playerViewDistance * 16);
                 double d1 = vec3_dx * vec3_dx + vec3_dz * vec3_dz; // Paper
                 double d2 = d * d;
                 // Paper start - Configurable entity tracking range by Y
