From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Tue, 13 May 2025 18:53:12 -0500
Subject: [PATCH] Packet obfuscation and reduction


diff --git a/net/minecraft/server/level/ServerPlayer.java b/net/minecraft/server/level/ServerPlayer.java
index 2a483220da5076987f1e17ec0fe3a93e30f23054..76d7cc0fbaac1c762679556ecf6a698613559f55 100644
--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -394,6 +394,7 @@ public class ServerPlayer extends Player implements ca.spottedleaf.moonrise.patc
     public @Nullable String clientBrandName = null; // Paper - Brand support
     public org.bukkit.event.player.PlayerQuitEvent.QuitReason quitReason = null; // Paper - Add API for quit reason; there are a lot of changes to do if we change all methods leading to the event
     public boolean smoothWorldTeleport = false; // Slice
+    public boolean emitsTickEndEvents; // Slice
 
     // Paper start - rewrite chunk system
     private ca.spottedleaf.moonrise.patches.chunk_system.player.RegionizedPlayerChunkLoader.PlayerChunkLoaderData chunkLoader;
diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 1777d0ca112e983166d14da0e6befebf257b7fee..68314cddb47a67a737ccfe9808d6416766fcc952 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3597,6 +3597,10 @@ public class ServerGamePacketListenerImpl
         }
 
         this.receivedMovementThisTick = false;
+
+        if (player.emitsTickEndEvents) {
+            new io.papermc.paper.event.packet.ClientTickEndEvent(player.getBukkitEntity()).callEvent();
+        }
     }
 
     private void handlePlayerKnownMovement(Vec3 movement) {
