From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:40:21 -0600
Subject: [PATCH] Mutton equivalent to Beef


diff --git a/net/minecraft/world/food/Foods.java b/net/minecraft/world/food/Foods.java
index eaf897ec7cbf6642ce61d0b15db240b57c109497..e334df5aa90d07eebe934e1d9307278588557d67 100644
--- a/net/minecraft/world/food/Foods.java
+++ b/net/minecraft/world/food/Foods.java
@@ -24,7 +24,7 @@ public class Foods {
         .saturationModifier(FoodConstants.FOOD_SATURATION_NORMAL)
         .build();
     public static final FoodProperties COOKED_COD = new FoodProperties.Builder().nutrition(5).saturationModifier(FoodConstants.FOOD_SATURATION_NORMAL).build();
-    public static final FoodProperties COOKED_MUTTON = new FoodProperties.Builder().nutrition(6).saturationModifier(FoodConstants.FOOD_SATURATION_GOOD).build();
+    public static final FoodProperties COOKED_MUTTON = new FoodProperties.Builder().nutrition(8).saturationModifier(FoodConstants.FOOD_SATURATION_GOOD).build(); // Slice
     public static final FoodProperties COOKED_PORKCHOP = new FoodProperties.Builder()
         .nutrition(8)
         .saturationModifier(FoodConstants.FOOD_SATURATION_GOOD)
@@ -57,7 +57,7 @@ public class Foods {
         .build();
     public static final FoodProperties MELON_SLICE = new FoodProperties.Builder().nutrition(2).saturationModifier(FoodConstants.FOOD_SATURATION_LOW).build();
     public static final FoodProperties MUSHROOM_STEW = stew(6).build();
-    public static final FoodProperties MUTTON = new FoodProperties.Builder().nutrition(2).saturationModifier(FoodConstants.FOOD_SATURATION_LOW).build();
+    public static final FoodProperties MUTTON = new FoodProperties.Builder().nutrition(3).saturationModifier(FoodConstants.FOOD_SATURATION_LOW).build(); // Slice
     public static final FoodProperties POISONOUS_POTATO = new FoodProperties.Builder()
         .nutrition(2)
         .saturationModifier(FoodConstants.FOOD_SATURATION_LOW)
