From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:41:46 -0600
Subject: [PATCH] Override inWater


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 4c55d6eb4895f270f151298f0c4133f1c35afca4..3efa1c91a0a2fa60422dfa853e875af845184e00 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -400,6 +400,7 @@ public abstract class Entity
     public boolean saveable = true; // Slice
     public boolean vanished; // Slice
     public int trackingRange; // Slice
+    public boolean inWaterOverride; // Slice
 
     public void inactiveTick() {
     }
@@ -2131,7 +2132,7 @@ public abstract class Entity
     }
 
     public boolean isInWater() {
-        return this.wasTouchingWater;
+        return this.wasTouchingWater || this.inWaterOverride; // Slice
     }
 
     public boolean isInRain() {
@@ -2276,7 +2277,7 @@ public abstract class Entity
     }
 
     public boolean isEyeInFluid(final TagKey<Fluid> type) {
-        return this.fluidInteraction.isEyeInFluid(type);
+        return inWaterOverride || this.fluidInteraction.isEyeInFluid(type); // Slice
     }
 
     public boolean isInLava() {
