From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sun, 3 Aug 2025 13:13:11 -0500
Subject: [PATCH] Make Entity navigation public


diff --git a/net/minecraft/world/entity/Mob.java b/net/minecraft/world/entity/Mob.java
index d61eccf5a0e58845de22df98d9aca4957e645af6..2bd0ecae8701396ea320e72ad0e039620859b2d6 100644
--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -121,7 +121,7 @@ public abstract class Mob extends LivingEntity implements EquipmentUser, Leashab
     protected MoveControl moveControl;
     protected JumpControl jumpControl;
     private final BodyRotationControl bodyRotationControl;
-    protected PathNavigation navigation;
+    public PathNavigation navigation; // Slice
     public GoalSelector goalSelector;
     public @Nullable net.minecraft.world.entity.ai.goal.FloatGoal goalFloat; // Paper - Allow nerfed mobs to jump and float
     public GoalSelector targetSelector;
