From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:42:05 -0600
Subject: [PATCH] Make Entity navigation public


diff --git a/net/minecraft/world/entity/Mob.java b/net/minecraft/world/entity/Mob.java
index c2dd2da4bfb7d65ad2730ba1ce771e2b488beda4..b95f43322f7c7f433fa4e17639d95213dd265420 100644
--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -138,7 +138,7 @@ public abstract class Mob extends LivingEntity implements Targeting, EquipmentUs
     protected MoveControl moveControl;
     protected JumpControl jumpControl;
     private final BodyRotationControl bodyRotationControl;
-    protected PathNavigation navigation;
+    public PathNavigation navigation; // Slice
     protected final GoalSelector goalSelector;
     public net.minecraft.world.entity.ai.goal.@Nullable FloatGoal goalFloat; // Paper - Allow nerfed mobs to jump and float
     public GoalSelector targetSelector;
