From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:41:46 -0600
Subject: [PATCH] Override inWater


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index b6cddcabc2c464fe3cf9a4f120dbb732cab21008..c410e0aba82ee48e633d3b9f9733a14716bb5cc5 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -386,6 +386,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     public boolean saveable = true; // Slice
     public boolean vanished; // Slice
     public int trackingRange; // Slice
+    public boolean inWaterOverride; // Slice
 
     public void inactiveTick() {
     }
@@ -1999,7 +2000,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     }
 
     public boolean isInWater() {
-        return this.wasTouchingWater;
+        return this.wasTouchingWater || this.inWaterOverride; // Slice
     }
 
     public boolean isInRain() {
@@ -2168,7 +2169,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     }
 
     public boolean isEyeInFluid(TagKey<Fluid> fluidTag) {
-        return this.fluidOnEyes.contains(fluidTag);
+        return inWaterOverride || this.fluidOnEyes.contains(fluidTag);
     }
 
     public boolean isInLava() {
