From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 15 Jan 2026 09:42:49 -0600
Subject: [PATCH] Add Vanish, settable inv locations


diff --git a/net/minecraft/server/ServerScoreboard.java b/net/minecraft/server/ServerScoreboard.java
index 7ffd683c42557052d7bfa963d87e68f4e8f62d6c..43afb8435476e727075976138c28dd7a1c20f4a0 100644
--- a/net/minecraft/server/ServerScoreboard.java
+++ b/net/minecraft/server/ServerScoreboard.java
@@ -196,6 +196,7 @@ public class ServerScoreboard extends Scoreboard {
 
     @Override
     public void onTeamChanged(final PlayerTeam team) {
+        if (!team.broadcastsChanges()) return; // Slice
         super.onTeamChanged(team);
         this.broadcastAll(ClientboundSetPlayerTeamPacket.createAddOrModifyPacket(team, false)); // CraftBukkit
         this.updateTeamWaypoints(team);
diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index 4f68ddc1da21dc665619df3d404396f35ab68bd6..264c14295e40260be696f0a4e6adbf30286897c1 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -1778,6 +1778,7 @@ public abstract class Player extends Avatar implements ContainerUser {
 
     @Override
     protected Entity.MovementEmission getMovementEmission() {
+        if (vanished) return MovementEmission.NONE; // Slice
         return this.abilities.flying || this.onGround() && this.isDiscrete() ? Entity.MovementEmission.NONE : Entity.MovementEmission.ALL;
     }
 
diff --git a/net/minecraft/world/scores/PlayerTeam.java b/net/minecraft/world/scores/PlayerTeam.java
index 2bdbb8451974f0b8d6e94d390ac735f4ccffd954..f12a28cdb27dd43b2635b9f4076909056e3d75b9 100644
--- a/net/minecraft/world/scores/PlayerTeam.java
+++ b/net/minecraft/world/scores/PlayerTeam.java
@@ -36,6 +36,7 @@ public class PlayerTeam extends Team {
     private Optional<TeamColor> color = Optional.empty();
     private Team.CollisionRule collisionRule = Team.CollisionRule.ALWAYS;
     private final Style displayNameStyle;
+    private boolean broadcastChanges = true; // Slice
 
     public PlayerTeam(final Scoreboard scoreboard, final String name) {
         this.scoreboard = scoreboard;
@@ -60,6 +61,16 @@ public class PlayerTeam extends Team {
         );
     }
 
+    // Slice start
+    public boolean broadcastsChanges() {
+        return broadcastChanges;
+    }
+
+    public void setBroadcastChanges(boolean broadcastChanges) {
+        this.broadcastChanges = broadcastChanges;
+    }
+    // Slice end
+
     public Scoreboard getScoreboard() {
         return this.scoreboard;
     }
