From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 21 Jan 2026 10:33:02 -0600
Subject: [PATCH] add EntityTryLungeEvent


diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index b5e24e1d024312e058b46fd00d7be232d4a1dd5e..8dadc5f96d1062fd4c66c8f4340d30b25632ebcd 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -2210,7 +2210,7 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
     }
 
     public void lungeForwardMaybe() {
-        if (this.level() instanceof ServerLevel serverLevel) {
+        if (this.level() instanceof ServerLevel serverLevel && new io.papermc.paper.event.entity.EntityTryLungeEvent(getBukkitLivingEntity()).callEvent()) {
             EnchantmentHelper.doLungeEffects(serverLevel, this);
         }
     }
