From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 21 Jan 2026 10:33:02 -0600
Subject: [PATCH] add EntityTryLungeEvent


diff --git a/net/minecraft/world/item/enchantment/Enchantment.java b/net/minecraft/world/item/enchantment/Enchantment.java
index 0d981b86fa7ee8713e2a419fc2f377defbc2ed9c..b192c0b1d1607a074882fe3ba9e2f86be292e125 100644
--- a/net/minecraft/world/item/enchantment/Enchantment.java
+++ b/net/minecraft/world/item/enchantment/Enchantment.java
@@ -356,6 +356,7 @@ public record Enchantment(Component description, Enchantment.EnchantmentDefiniti
             e -> {
                 int level = enchantmentLevel;
                 if (user instanceof LivingEntity livingEntity) {
+                    if (!new io.papermc.paper.event.entity.EntityTryLungeEvent(livingEntity.getBukkitLivingEntity()).callEvent()) return; // Slice
                     final io.papermc.paper.event.entity.EntityLungeEvent event = new io.papermc.paper.event.entity.EntityLungeEvent(livingEntity.getBukkitLivingEntity(), level);
                     if (!event.callEvent()) {
                         return;
