From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sat, 9 Aug 2025 17:57:21 -0500
Subject: [PATCH] Force random tick Leaf Blocks


diff --git a/net/minecraft/world/level/block/LeavesBlock.java b/net/minecraft/world/level/block/LeavesBlock.java
index 9dfb843dc5a5d94ac52e2ee3249182087ded40f1..568610900bd4d5e7e06e0fc9bb568525a5f70b79 100644
--- a/net/minecraft/world/level/block/LeavesBlock.java
+++ b/net/minecraft/world/level/block/LeavesBlock.java
@@ -53,7 +53,7 @@ public abstract class LeavesBlock extends Block implements SimpleWaterloggedBloc
 
     @Override
     protected boolean isRandomlyTicking(BlockState state) {
-        return state.getValue(DISTANCE) == 7 && !state.getValue(PERSISTENT);
+        return true; //state.getValue(DISTANCE) == 7 && !state.getValue(PERSISTENT);
     }
 
     @Override
