From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Mon, 26 Jan 2026 15:47:13 -0600
Subject: [PATCH] Loka-specific spear changes


diff --git a/net/minecraft/world/item/Items.java b/net/minecraft/world/item/Items.java
index f7d7bd4f06358183b1ed54bfef4c4db822ebe0f9..97ca78ec0a51ef564b20127e48824855c0e83425 100644
--- a/net/minecraft/world/item/Items.java
+++ b/net/minecraft/world/item/Items.java
@@ -1668,22 +1668,22 @@ public class Items {
             .component(DataComponents.BREAK_SOUND, SoundEvents.SHIELD_BREAK)
     );
     public static final Item WOODEN_SPEAR = registerItem(
-        ItemIds.WOODEN_SPEAR, new Item.Properties().spear(ToolMaterial.WOOD, 0.65F, 0.7F, 0.75F, 5.0F, 14.0F, 10.0F, 5.1F, 15.0F, 4.6F)
+        ItemIds.WOODEN_SPEAR, new Item.Properties().spear(ToolMaterial.WOOD, 1.15F, 0.7F, 0.75F, 5.0F, 14.0F, 10.0F, 5.1F, 15.0F, 4.6F) // Slice - make all spears diamond and below share swingDurationSeconds
     );
     public static final Item STONE_SPEAR = registerItem(
-        ItemIds.STONE_SPEAR, new Item.Properties().spear(ToolMaterial.STONE, 0.75F, 0.82F, 0.7F, 4.5F, 13.0F, 9.0F, 5.1F, 13.75F, 4.6F)
+        ItemIds.STONE_SPEAR, new Item.Properties().spear(ToolMaterial.STONE, 1.15F, 0.82F, 0.7F, 4.5F, 13.0F, 9.0F, 5.1F, 13.75F, 4.6F) // Slice - make all spears diamond and below share swingDurationSeconds
     );
     public static final Item COPPER_SPEAR = registerItem(
-        ItemIds.COPPER_SPEAR, new Item.Properties().spear(ToolMaterial.COPPER, 0.85F, 0.82F, 0.65F, 4.0F, 12.0F, 8.25F, 5.1F, 12.5F, 4.6F)
+        ItemIds.COPPER_SPEAR, new Item.Properties().spear(ToolMaterial.COPPER, 1.15F, 0.82F, 0.65F, 4.0F, 12.0F, 8.25F, 5.1F, 12.5F, 4.6F) // Slice - make all spears diamond and below share swingDurationSeconds
     );
     public static final Item IRON_SPEAR = registerItem(
-        ItemIds.IRON_SPEAR, new Item.Properties().spear(ToolMaterial.IRON, 0.95F, 0.95F, 0.6F, 2.5F, 11.0F, 6.75F, 5.1F, 11.25F, 4.6F)
+        ItemIds.IRON_SPEAR, new Item.Properties().spear(ToolMaterial.IRON, 1.15F, 0.95F, 0.6F, 2.5F, 11.0F, 6.75F, 5.1F, 11.25F, 4.6F) // Slice - make all spears diamond and below share swingDurationSeconds
     );
     public static final Item GOLDEN_SPEAR = registerItem(
-        ItemIds.GOLDEN_SPEAR, new Item.Properties().spear(ToolMaterial.GOLD, 0.95F, 0.7F, 0.7F, 3.5F, 13.0F, 8.5F, 5.1F, 13.75F, 4.6F)
+        ItemIds.GOLDEN_SPEAR, new Item.Properties().spear(ToolMaterial.GOLD, 1.15F, 0.7F, 0.7F, 3.5F, 13.0F, 8.5F, 5.1F, 13.75F, 4.6F) // Slice - make all spears diamond and below share swingDurationSeconds
     );
     public static final Item DIAMOND_SPEAR = registerItem(
-        ItemIds.DIAMOND_SPEAR, new Item.Properties().spear(ToolMaterial.DIAMOND, 1.05F, 1.075F, 0.5F, 3.0F, 10.0F, 6.5F, 5.1F, 10.0F, 4.6F)
+        ItemIds.DIAMOND_SPEAR, new Item.Properties().spear(ToolMaterial.DIAMOND, 1.15F, 1.075F, 0.5F, 3.0F, 10.0F, 6.5F, 5.1F, 10.0F, 4.6F) // Slice
     );
     public static final Item NETHERITE_SPEAR = registerItem(
         ItemIds.NETHERITE_SPEAR, new Item.Properties().spear(ToolMaterial.NETHERITE, 1.15F, 1.2F, 0.4F, 2.5F, 9.0F, 5.5F, 5.1F, 8.75F, 4.6F).fireResistant()
