From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 28 Jan 2026 10:23:02 -0600
Subject: [PATCH] Force equipment update before piercing weapon attack


diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index dd6dce15e04721f0de3ce1efb76a53144925773e..be4da58c4aefd8b99eeb2d00337834c2d6f49673 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1924,6 +1924,7 @@ public class ServerGamePacketListenerImpl
 
                         PiercingWeapon piercingWeapon = itemInHand.get(DataComponents.PIERCING_WEAPON);
                         if (piercingWeapon != null) {
+                            if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
                             piercingWeapon.attack(this.player, EquipmentSlot.MAINHAND);
                         }
 
