From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 13 Aug 2025 15:59:59 -0500
Subject: [PATCH] Disable entity id spam


diff --git a/net/minecraft/world/entity/EntityType.java b/net/minecraft/world/entity/EntityType.java
index cdf71340278e05e58bbb4ed51a432336de46e549..cccdcbd9a0096f2deab9af518d0e4cf220453b68 100644
--- a/net/minecraft/world/entity/EntityType.java
+++ b/net/minecraft/world/entity/EntityType.java
@@ -1394,7 +1394,9 @@ public class EntityType<T extends Entity> implements FeatureElement, EntityTypeT
                 entity.load(input);
             },
             // Paper end - Don't fire sync event during generation
-            () -> LOGGER.warn("Skipping Entity with id {}", input.getStringOr("id", "[invalid]"))
+            () -> {
+//                LOGGER.warn("Skipping Entity with id {}", input.getStringOr("id", "[invalid]"));
+            }
         );
     }
 
