From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 28 Jan 2026 10:23:02 -0600
Subject: [PATCH] Force equipment update before piercing weapon attack


diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 19a457003eb62f448539783e71ce1faf57d03edb..183f74e2b8af94b90a80b9c09608537008cf46cd 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1952,6 +1952,7 @@ public class ServerGamePacketListenerImpl
 
                         PiercingWeapon piercingWeapon = itemInHand.get(DataComponents.PIERCING_WEAPON);
                         if (piercingWeapon != null) {
+                            if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
                             piercingWeapon.attack(this.player, EquipmentSlot.MAINHAND);
                         }
 
