From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Wed, 28 Jan 2026 10:23:02 -0600
Subject: [PATCH] Force equipment update before piercing weapon attack


diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index c93869b93f1dd39f0a7ba627c1fdfc3567988ce3..72e1674f33a83252cb47ba46ec8cb40d6dcd7639 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1923,6 +1923,7 @@ public class ServerGamePacketListenerImpl
 
                         PiercingWeapon piercingWeapon = itemInHand.get(DataComponents.PIERCING_WEAPON);
                         if (piercingWeapon != null) {
+                            if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
                             piercingWeapon.attack(this.player, EquipmentSlot.MAINHAND);
                         }
 
