From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 19 Feb 2026 09:07:40 -0600
Subject: [PATCH] Add extraHitboxScale to entities


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 3efa1c91a0a2fa60422dfa853e875af845184e00..6fe4e12088d4ffeae45afb9d8ad42fae71bfe4ee 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -401,6 +401,7 @@ public abstract class Entity
     public boolean vanished; // Slice
     public int trackingRange; // Slice
     public boolean inWaterOverride; // Slice
+    public float extraHitboxAmount; // Slice
 
     public void inactiveTick() {
     }
@@ -4601,7 +4602,9 @@ public abstract class Entity
         return this.bb;
     }
 
-    public final void setBoundingBox(final AABB bb) {
+    public final void setBoundingBox(AABB bb) {
+        bb = bb.inflate(extraHitboxAmount); // Slice
+
         // CraftBukkit start - block invalid bounding boxes
         double minX = bb.minX,
                 minY = bb.minY,
