From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Thu, 19 Feb 2026 09:07:40 -0600
Subject: [PATCH] Add extraHitboxScale to entities


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index c410e0aba82ee48e633d3b9f9733a14716bb5cc5..f445de3ac547779a2b9efa1c2f41da9aff800d53 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -387,6 +387,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     public boolean vanished; // Slice
     public int trackingRange; // Slice
     public boolean inWaterOverride; // Slice
+    public float extraHitboxAmount; // Slice
 
     public void inactiveTick() {
     }
@@ -4473,6 +4474,8 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     }
 
     public final void setBoundingBox(AABB bb) {
+        bb = bb.inflate(extraHitboxAmount);
+
         // CraftBukkit start - block invalid bounding boxes
         double minX = bb.minX,
                 minY = bb.minY,
