From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Sat, 18 Apr 2026 18:28:10 -0500
Subject: [PATCH] Entity Visibility Buckets


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index f445de3ac547779a2b9efa1c2f41da9aff800d53..2caf6128ea5f5cb4915a279eccf9d592e0c9c139 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -388,6 +388,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     public int trackingRange; // Slice
     public boolean inWaterOverride; // Slice
     public float extraHitboxAmount; // Slice
+    public long visibilityBuckets = 0; // Slice
 
     public void inactiveTick() {
     }
