From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: cryptite <cryptite@gmail.com>
Date: Thu, 18 Jun 2026 08:09:14 -0500
Subject: [PATCH] Mute block-attached entity error


diff --git a/net/minecraft/world/entity/decoration/BlockAttachedEntity.java b/net/minecraft/world/entity/decoration/BlockAttachedEntity.java
index 10500f4979331bd676abe59f2fbfff03aef20910..31ce66b7f0a8fe5e196b2d4ecbdbdd23a3b5ec6e 100755
--- a/net/minecraft/world/entity/decoration/BlockAttachedEntity.java
+++ b/net/minecraft/world/entity/decoration/BlockAttachedEntity.java
@@ -166,7 +166,7 @@ public abstract class BlockAttachedEntity extends Entity {
         if (blockPos != null && blockPos.closerThan(this.blockPosition(), 16.0)) {
             this.pos = blockPos;
         } else {
-            LOGGER.error("Block-attached entity at invalid position: {}", blockPos);
+//            LOGGER.error("Block-attached entity at invalid position: {}", blockPos);
         }
     }
 
