From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: cryptite <cryptite@gmail.com>
Date: Thu, 18 Jun 2026 08:09:14 -0500
Subject: [PATCH] Mute block-attached entity error


diff --git a/net/minecraft/world/entity/decoration/BlockAttachedEntity.java b/net/minecraft/world/entity/decoration/BlockAttachedEntity.java
index 914049eec8b5a4b5190c497b09e398bc23ac4fef..2fecb12c74897fd7c6aeacccd69ef46f8ad4aea9 100644
--- a/net/minecraft/world/entity/decoration/BlockAttachedEntity.java
+++ b/net/minecraft/world/entity/decoration/BlockAttachedEntity.java
@@ -168,7 +168,7 @@ public abstract class BlockAttachedEntity extends Entity {
         if (storedPos != null && storedPos.closerThan(this.blockPosition(), 16.0)) {
             this.pos = storedPos;
         } else {
-            LOGGER.error("Block-attached entity at invalid position: {}", storedPos);
+//            LOGGER.error("Block-attached entity at invalid position: {}", storedPos); // Slice
         }
     }
 
