--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3902,20 +_,4 @@
      * @return the result of this method, holding leftovers and spawned items.
      */
     PlayerGiveResult give(Collection<ItemStack> items, boolean dropIfFull);
-
-    /**
-     * Get the score that shows in the death screen of the player.
-     * <p>This amount is added to when the player gains experience.</p>
-     *
-     * @return Death screen score of player
-     */
-    int getDeathScreenScore();
-
-    /**
-     * Set the score that shows in the death screen of the player.
-     * <p>This amount is added to when the player gains experience.</p>
-     *
-     * @param score New death screen score of player
-     */
-    void setDeathScreenScore(int score);
 }
