--- a/src/main/java/io/papermc/paper/datacomponent/PaperDataComponentType.java
+++ b/src/main/java/io/papermc/paper/datacomponent/PaperDataComponentType.java
@@ -77,9 +_,7 @@
         if (adapter == null) {
             throw new IllegalArgumentException("No adapter found for " + key);
         }
-        if (adapter.isUnimplemented()) {
-            return new Unimplemented<>(key, type, adapter);
-        } else if (adapter.isValued()) {
+        if (adapter.isValued()) {
             return new ValuedImpl<>(key, type, adapter);
         } else {
             return new NonValuedImpl<>(key, type, adapter);
@@ -100,17 +_,6 @@
     public static final class ValuedImpl<T, NMS> extends PaperDataComponentType<T, NMS> implements Valued<T> {
 
         ValuedImpl(
-            final NamespacedKey key,
-            final net.minecraft.core.component.DataComponentType<NMS> type,
-            final DataComponentAdapter<NMS, T> adapter
-        ) {
-            super(key, type, adapter);
-        }
-    }
-
-    public static final class Unimplemented<T, NMS> extends PaperDataComponentType<T, NMS> {
-
-        public Unimplemented(
             final NamespacedKey key,
             final net.minecraft.core.component.DataComponentType<NMS> type,
             final DataComponentAdapter<NMS, T> adapter
