--- a/src/main/java/io/papermc/paper/datacomponent/item/PaperCustomModelData.java
+++ b/src/main/java/io/papermc/paper/datacomponent/item/PaperCustomModelData.java
@@ -40,7 +_,7 @@
 
     @Override
     public List<Color> colors() {
-        return MCUtil.transformUnmodifiable(this.impl.colors(), color -> Color.fromRGB(color & 0x00FFFFFF)); // skip alpha channel
+        return MCUtil.transformUnmodifiable(this.impl.colors(), Color::fromRGB);
     }
 
     static final class BuilderImpl implements CustomModelData.Builder {
